<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Exposes and makes available data from the cBioPortal web resources â€¢ cBioPortalData</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Exposes and makes available data from the cBioPortal web resources">
<meta property="og:description" content="The cBioPortalData R package accesses study datasets from the
    cBio Cancer Genomics Portal. It accesses the data either from
    the pre-packaged zip / tar files or from the API interface that was
    recently implemented by the cBioPortal Data Team. The package can provide
    data in either tabular format or with MultiAssayExperiment object that uses
    familiar Bioconductor data representations.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">cBioPortalData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.13.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/cBioPortalData.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/cBioPortalDataErrors.html">cBioPortalData: Data Build Errors</a>
    </li>
    <li>
      <a href="articles/cBioPortalRClient.html">cBioPortalData: API Reference Guide for Devs</a>
    </li>
    <li>
      <a href="articles/cgdsrMigration.html">cgdsr to cBioPortalData: Migration Tutorial</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/cBioPortalData/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="cbioportaldata">cBioPortalData<a class="anchor" aria-label="anchor" href="#cbioportaldata"></a>
</h1></div>
<!-- start badges here -->
<p><a href="https://bioconductor.org/checkResults/release/bioc-LATEST/cBioPortalData" class="external-link"><img src="http://www.bioconductor.org/shields/build/release/bioc/cBioPortalData.svg" alt="BioC status"></a> <a href="https://www.bioconductor.org/packages/release/bioc/html/cBioPortalData.html#archives" class="external-link"><img src="http://www.bioconductor.org/shields/availability/release/cBioPortalData.svg" alt="Platforms"></a> <a href="https://bioconductor.org/packages/stats/bioc/cBioPortalData/" class="external-link"><img src="https://www.bioconductor.org/shields/downloads/release/cBioPortalData.svg" alt="Downloads"></a> <!-- end badges here --></p>
<div class="section level2">
<h2 id="cbioportal-data-and-multiassayexperiment">cBioPortal data and MultiAssayExperiment<a class="anchor" aria-label="anchor" href="#cbioportal-data-and-multiassayexperiment"></a>
</h2>
<div class="section level3">
<h3 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h3>
<p>The <code>cBioPortalData</code> R package aims to import cBioPortal datasets as <a href="http://bioconductor.org/packages/MultiAssayExperiment/" class="external-link">MultiAssayExperiment</a> objects into Bioconductor. Some of the features of the package include:</p>
<ol style="list-style-type: decimal">
<li>The use of the <code>MultiAssayExperiment</code> integrative container for coordinating and representing the data.</li>
<li>The data container explicitly links all assays to the patient clinical/pathological data.</li>
<li>With a <a href="https://github.com/waldronlab/MultiAssayExperiment/wiki/MultiAssayExperiment-API" class="external-link">flexible API</a>, <code>MultiAssayExperiment</code> provides harmonized subsetting and reshaping into convenient wide and long formats.</li>
<li>The package provides datasets from both the API and the saved packaged data.</li>
<li>It also provides automatic local caching, thanks to <a href="https://bioconductor.org/packages/BiocFileCache/" class="external-link">BiocFileCache</a>.</li>
</ol>
</div>
</div>
<div class="section level2">
<h2 id="multiassayexperiment-cheatsheet">MultiAssayExperiment Cheatsheet<a class="anchor" aria-label="anchor" href="#multiassayexperiment-cheatsheet"></a>
</h2>
<p><a href="https://github.com/waldronlab/cheatsheets/blob/master/MultiAssayExperiment_QuickRef.pdf" class="external-link"> <img src="https://raw.githubusercontent.com/waldronlab/cheatsheets/master/pngs/MultiAssayExperiment_QuickRef.png" width="989" height="1091"></a></p>
</div>
<div class="section level2">
<h2 id="quick-start">Quick Start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h2>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>To install from Bioconductor (recommended for most users, this will install the release or development version corresponding to your version of Bioconductor):</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"cBioPortalData"</span><span class="op">)</span></span></code></pre></div>
<p>To install from GitHub (for bleeding-edge, not generally necessary because changes here are also pushed to <a href="https://www.bioconductor.org/developers/how-to/useDevel/" class="external-link">bioc-devel</a>. Installing the development version and generally requires running <a href="https://www.bioconductor.org/developers/how-to/useDevel/" class="external-link">bioc-devel</a>)</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"cBioPortalData"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"waldronlab/cBioPortalData"</span><span class="op">)</span></span></code></pre></div>
<p>To load the package:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">cBioPortalData</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a>
</h2>
<p><code>cBioPortalData</code> is a work in progress due to changes in data curation and cBioPortal API specification. Users can view the <code>data(studiesTable)</code> dataset to get an overview of the studies that are available and currently building as <code>MultiAssayExperiment</code> representations. About 98 % of the studies via the API (<code>api_build</code>) and 73 % of the package studies (<code>pack_build</code>) are building, these include additional datasets that were not previously available. Feel free to file an issue to request prioritization of fixing any of the remaining datasets.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cbio</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cBioPortal.html">cBioPortal</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">studies</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cBioPortal.html">getStudies</a></span><span class="op">(</span><span class="va">cbio</span>, buildReport <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">studies</span><span class="op">$</span><span class="va">api_build</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; FALSE  TRUE </span></span>
<span><span class="co">#&gt;     5   308</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">studies</span><span class="op">$</span><span class="va">pack_build</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; FALSE  TRUE </span></span>
<span><span class="co">#&gt;    86   227</span></span></code></pre></div>
<div class="section level3">
<h3 id="api-service">API Service<a class="anchor" aria-label="anchor" href="#api-service"></a>
</h3>
<p>Flexible and granular access to cBioPortal data from <code>cbioportal.org/api</code>. This option is best used with a particular gene panel of interest. It allows users to download sections of the data with molecular profile and gene panel combinations within a study.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gbm</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cBioPortalData.html">cBioPortalData</a></span><span class="op">(</span>api <span class="op">=</span> <span class="va">cbio</span>, by <span class="op">=</span> <span class="st">"hugoGeneSymbol"</span>, studyId <span class="op">=</span> <span class="st">"gbm_tcga"</span>,</span>
<span>    genePanelId <span class="op">=</span> <span class="st">"IMPACT341"</span>,</span>
<span>    molecularProfileIds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gbm_tcga_rppa"</span>, <span class="st">"gbm_tcga_mrna"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gbm</span></span>
<span><span class="co">#&gt; A MultiAssayExperiment object of 2 listed</span></span>
<span><span class="co">#&gt;  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">#&gt;  Containing an ExperimentList class object of length 2:</span></span>
<span><span class="co">#&gt;  [1] gbm_tcga_rppa: SummarizedExperiment with 67 rows and 244 columns</span></span>
<span><span class="co">#&gt;  [2] gbm_tcga_mrna: SummarizedExperiment with 336 rows and 401 columns</span></span>
<span><span class="co">#&gt; Functionality:</span></span>
<span><span class="co">#&gt;  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">#&gt;  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">#&gt;  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">#&gt;  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">#&gt;  exportClass() - save data to flat files</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="packaged-data-service">Packaged Data Service<a class="anchor" aria-label="anchor" href="#packaged-data-service"></a>
</h3>
<p>This function will download a dataset from the <code>cbioportal.org/datasets</code> website as a packaged tarball and serve it to users as a <code>MultiAssayExperiment</code> object. This option is good for users who are interested in obtaining all the data for a particular study.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">acc</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cBioDataPack.html">cBioDataPack</a></span><span class="op">(</span><span class="st">"acc_tcga"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">acc</span></span>
<span><span class="co">#&gt; A MultiAssayExperiment object of 11 listed</span></span>
<span><span class="co">#&gt;  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">#&gt;  Containing an ExperimentList class object of length 11:</span></span>
<span><span class="co">#&gt;  [1] cna_hg19.seg: RaggedExperiment with 16080 rows and 90 columns</span></span>
<span><span class="co">#&gt;  [2] CNA: SummarizedExperiment with 24776 rows and 90 columns</span></span>
<span><span class="co">#&gt;  [3] linear_CNA: SummarizedExperiment with 24776 rows and 90 columns</span></span>
<span><span class="co">#&gt;  [4] methylation_hm450: SummarizedExperiment with 15755 rows and 80 columns</span></span>
<span><span class="co">#&gt;  [5] mutations_extended: RaggedExperiment with 20166 rows and 90 columns</span></span>
<span><span class="co">#&gt;  [6] mutations_mskcc: RaggedExperiment with 20166 rows and 90 columns</span></span>
<span><span class="co">#&gt;  [7] RNA_Seq_v2_expression_median: SummarizedExperiment with 20531 rows and 79 columns</span></span>
<span><span class="co">#&gt;  [8] RNA_Seq_v2_mRNA_median_all_sample_Zscores: SummarizedExperiment with 20531 rows and 79 columns</span></span>
<span><span class="co">#&gt;  [9] RNA_Seq_v2_mRNA_median_Zscores: SummarizedExperiment with 20440 rows and 79 columns</span></span>
<span><span class="co">#&gt;  [10] rppa_Zscores: SummarizedExperiment with 191 rows and 46 columns</span></span>
<span><span class="co">#&gt;  [11] rppa: SummarizedExperiment with 192 rows and 46 columns</span></span>
<span><span class="co">#&gt; Functionality:</span></span>
<span><span class="co">#&gt;  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">#&gt;  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">#&gt;  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">#&gt;  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">#&gt;  exportClass() - save data to flat files</span></span></code></pre></div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://www.bioconductor.org/packages/cBioPortalData" class="external-link">View on Bioconductor</a></li>
<li><a href="https://github.com/waldronlab/cBioPortalData/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/waldronlab/cBioPortalData/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/AGPL-3" class="external-link">AGPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing cBioPortalData</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Levi Waldron <br><small class="roles"> Author </small>  </li>
<li>Marcel Ramos <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-3242-0582" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Levi Waldron, Marcel Ramos.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
