<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>The R interface to the cBioPortal API Data Service — cBioPortal • cBiocPortalData</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/spacelab/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="The R interface to the cBioPortal API Data Service — cBioPortal" />
<meta property="og:description" content="This function gives users access to the cBioPortal data with an
    API representation invoked from `cBioPortal()`" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cBiocPortalData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Quick Reference</a>
</li>
<li>
  <a href="../articles/cBioPortalData.html">User Start Guide</a>
</li>
<li>
  <a href="../articles/cBioPortalRClient.html">API Reference Guide</a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    NEWS
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/waldronlab/cBioPortalData">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>The R interface to the cBioPortal API Data Service</h1>
    <small class="dont-index">Source: <a href='https://github.com/waldronlab/cBioPortalData/blob/master/R/cBioPortal.R'><code>R/cBioPortal.R</code></a></small>
    <div class="hidden name"><code>cBioPortal.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function gives users access to the cBioPortal data with an
    API representation invoked from `cBioPortal()`</p>
    </div>

    <pre class="usage"><span class='fu'>cBioPortal</span>()

<span class='fu'>getStudies</span>(<span class='no'>api</span>)

<span class='fu'>clinicalData</span>(<span class='no'>api</span>, <span class='kw'>studyId</span> <span class='kw'>=</span> <span class='fl'>NA_character_</span>)

<span class='fu'>molecularProfiles</span>(
  <span class='no'>api</span>,
  <span class='kw'>studyId</span> <span class='kw'>=</span> <span class='fl'>NA_character_</span>,
  <span class='kw'>projection</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"SUMMARY"</span>, <span class='st'>"ID"</span>, <span class='st'>"DETAILED"</span>, <span class='st'>"META"</span>)
)

<span class='fu'>molecularData</span>(
  <span class='no'>api</span>,
  <span class='kw'>molecularProfileId</span> <span class='kw'>=</span> <span class='fl'>NA_character_</span>,
  <span class='kw'>entrezGeneIds</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>sampleIds</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>check</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>
)

<span class='fu'>searchOps</span>(<span class='no'>api</span>, <span class='no'>keyword</span>)

<span class='fu'>geneTable</span>(<span class='no'>api</span>, <span class='no'>...</span>)

<span class='fu'>samplesInSampleLists</span>(<span class='no'>api</span>, <span class='kw'>sampleListIds</span> <span class='kw'>=</span> <span class='fl'>NA_character_</span>, <span class='kw'>check</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)

<span class='fu'>sampleLists</span>(<span class='no'>api</span>, <span class='kw'>studyId</span> <span class='kw'>=</span> <span class='fl'>NA_character_</span>)

<span class='fu'>allSamples</span>(<span class='no'>api</span>, <span class='kw'>studyId</span> <span class='kw'>=</span> <span class='fl'>NA_character_</span>)

<span class='fu'>genePanels</span>(<span class='no'>api</span>)

<span class='fu'>getGenePanel</span>(<span class='no'>api</span>, <span class='kw'>genePanelId</span> <span class='kw'>=</span> <span class='fl'>NA_character_</span>)

<span class='fu'>genePanelMolecular</span>(
  <span class='no'>api</span>,
  <span class='kw'>molecularProfileId</span> <span class='kw'>=</span> <span class='fl'>NA_character_</span>,
  <span class='kw'>sampleListId</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>sampleIds</span> <span class='kw'>=</span> <span class='kw'>NULL</span>
)

<span class='fu'>getGenePanelMolecular</span>(<span class='no'>api</span>, <span class='kw'>molecularProfileIds</span> <span class='kw'>=</span> <span class='fl'>NA_character_</span>, <span class='no'>sampleIds</span>)

<span class='fu'>getSampleInfo</span>(
  <span class='no'>api</span>,
  <span class='kw'>studyId</span> <span class='kw'>=</span> <span class='fl'>NA_character_</span>,
  <span class='kw'>sampleListIds</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>projection</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"SUMMARY"</span>, <span class='st'>"ID"</span>, <span class='st'>"DETAILED"</span>, <span class='st'>"META"</span>)
)

<span class='fu'>getDataByGenePanel</span>(
  <span class='no'>api</span>,
  <span class='kw'>studyId</span> <span class='kw'>=</span> <span class='fl'>NA_character_</span>,
  <span class='kw'>genePanelId</span> <span class='kw'>=</span> <span class='fl'>NA_character_</span>,
  <span class='kw'>molecularProfileId</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>sampleListId</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>check</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>api</th>
      <td><p>An API object of class `cBioPortal` from the `cBioPortal`
function</p></td>
    </tr>
    <tr>
      <th>studyId</th>
      <td><p>character(1) Indicates the "studyId" as taken from
`getStudies`</p></td>
    </tr>
    <tr>
      <th>projection</th>
      <td><p>character(default: "SUMMARY") Specify the projection
type for data retrieval for details see API documentation</p></td>
    </tr>
    <tr>
      <th>molecularProfileId</th>
      <td><p>character(1) Indicates a molecular profile ID</p></td>
    </tr>
    <tr>
      <th>entrezGeneIds</th>
      <td><p>numeric() A vector indicating entrez gene IDs</p></td>
    </tr>
    <tr>
      <th>sampleIds</th>
      <td><p>character() Sample identifiers</p></td>
    </tr>
    <tr>
      <th>check</th>
      <td><p>logical(1) Whether to check the inputs against values from the
API (i.e., for 'studyId', 'genePanelId', 'molecularProfileId', and
'sampleListId')</p></td>
    </tr>
    <tr>
      <th>keyword</th>
      <td><p>character(1) Keyword or pattern for searching through
available operations</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional arguments to lower level API functions</p></td>
    </tr>
    <tr>
      <th>sampleListIds</th>
      <td><p>character() A vector of 'sampleListId' as obtained from
`sampleLists`</p></td>
    </tr>
    <tr>
      <th>genePanelId</th>
      <td><p>character(1) Identifies the gene panel, as obtained
from the `genePanels` function</p></td>
    </tr>
    <tr>
      <th>sampleListId</th>
      <td><p>character(1) A sample list identifier as obtained from
`sampleLists()``</p></td>
    </tr>
    <tr>
      <th>molecularProfileIds</th>
      <td><p>character() A vector of molecular profile IDs</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>cBioPortal: An API object of class 'cBioPortal'</p>
<p>cBioPortalData: A data object of class 'MultiAssayExperiment'</p>
    <h2 class="hasAnchor" id="api-metadata"><a class="anchor" href="#api-metadata"></a>API Metadata</h2>

    

<p>* getStudies - Obtain a table of studies and associated metadata</p>

<p>* searchOps - Search through API operations with a keyword</p>

<p>* geneTable - Get a table of all genes by 'entrezGeneId' or
    'hugoGeneSymbol'</p>

<p>* sampleLists - obtain all `sampleListIds` for a particular `studyId`</p>

<p>* allSamples - obtain all samples within a particular `studyId`</p>

<p>* genePanels - Show all available gene panels</p>
    <h2 class="hasAnchor" id="patient-data"><a class="anchor" href="#patient-data"></a>Patient Data</h2>

    

<p>* clinicalData - Obtain clinical data for a particular study identifier
     ('studyId')</p>
    <h2 class="hasAnchor" id="molecular-profiles"><a class="anchor" href="#molecular-profiles"></a>Molecular Profiles</h2>

    

<p>* molecularProfiles - Produce a molecular profiles dataset for a given
     study identifier ('studyId')</p>

<p>* molecularData - Produce a dataset of molecular profile data based on
    `molecularProfileId`, `entrezGeneIds`, and `sampleIds`</p>
    <h2 class="hasAnchor" id="sample-data"><a class="anchor" href="#sample-data"></a>Sample Data</h2>

    

<p>* samplesInSampleLists - get all samples associated with a 'sampleListId'</p>

<p>* getSampleInfo - Obtain sample metadata for a particular `studyId` or
    `sampleListId`</p>
    <h2 class="hasAnchor" id="gene-panels"><a class="anchor" href="#gene-panels"></a>Gene Panels</h2>

    

<p>* getGenePanels - Obtain the gene panel for a particular 'genePanelId'</p>

<p>* genePanelMolecular - get gene panel data for a paricular
    `molecularProfileId` and `sampleListId` combination</p>

<p>* getGenePanelMolecular - get gene panel data for a combination of
    `molecularProfileId` and `sampleListId` vectors</p>

<p>* getDataByGenePanel - Download data for a gene panel and
    `molecularProfileId` combination, optionally a `sampleListId` can be
    provided.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='no'>cbio</span> <span class='kw'>&lt;-</span> <span class='fu'>cBioPortal</span>()

<span class='fu'>getStudies</span>(<span class='kw'>api</span> <span class='kw'>=</span> <span class='no'>cbio</span>)</div><div class='output co'>#&gt; <span style='color: #555555;'># A tibble: 280 x 13</span><span>
#&gt;    name  shortName description publicStudy pmid  citation groups status
#&gt;    </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span>       </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>    </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>   </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #555555;'> 1</span><span> Chol… Cholangi… Exome sequ… TRUE        2418… Chan-on… </span><span style='color: #555555;'>"</span><span>PUBL…      0
#&gt; </span><span style='color: #555555;'> 2</span><span> Chol… Cholangi… Exome Sequ… TRUE        2256… Ong et … </span><span style='color: #555555;'>"</span><span>PUBL…      0
#&gt; </span><span style='color: #555555;'> 3</span><span> Cuta… CTCL (Co… Whole-Exom… TRUE        2655… Da Silv… </span><span style='color: #555555;'>""</span><span>          0
#&gt; </span><span style='color: #555555;'> 4</span><span> Esop… ESCC (UC… Whole exom… TRUE        2468… Lin et … </span><span style='color: #555555;'>"</span><span>PUBL…      0
#&gt; </span><span style='color: #555555;'> 5</span><span> Oral… Head &amp; n… Comprehens… TRUE        2361… Pickeri… </span><span style='color: #555555;'>""</span><span>          0
#&gt; </span><span style='color: #555555;'> 6</span><span> Hepa… HCC (Ins… Whole-exom… TRUE        2582… Schulze… </span><span style='color: #555555;'>"</span><span>PUBL…      0
#&gt; </span><span style='color: #555555;'> 7</span><span> Uvea… UM (QIMR) Whole-geno… TRUE        2668… Johanss… </span><span style='color: #555555;'>"</span><span>PUBL…      0
#&gt; </span><span style='color: #555555;'> 8</span><span> Neur… NBL (AMC) Whole geno… TRUE        2236… Molenaa… </span><span style='color: #555555;'>"</span><span>PUBL…      0
#&gt; </span><span style='color: #555555;'> 9</span><span> Naso… NPC (Sin… Whole exom… TRUE        2495… Lin et … </span><span style='color: #555555;'>"</span><span>PUBL…      0
#&gt; </span><span style='color: #555555;'>10</span><span> Thym… TET (NCI) Whole exom… TRUE        2497… Petrini… </span><span style='color: #555555;'>"</span><span>PUBL…      0
#&gt; </span><span style='color: #555555;'># … with 270 more rows, and 5 more variables: importDate </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   allSampleCount </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span style='color: #555555;'>, studyId </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, cancerTypeId </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   referenceGenome </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span></div><div class='input'>
<span class='fu'>searchOps</span>(<span class='kw'>api</span> <span class='kw'>=</span> <span class='no'>cbio</span>, <span class='kw'>keyword</span> <span class='kw'>=</span> <span class='st'>"molecular"</span>)</div><div class='output co'>#&gt;  [1] "fetchGenePanelDataInMultipleMolecularProfilesUsingPOST"   
#&gt;  [2] "fetchGenericAssayDataInMultipleMolecularProfilesUsingPOST"
#&gt;  [3] "fetchGenericAssayDataInMolecularProfileUsingPOST"         
#&gt;  [4] "fetchMolecularDataInMultipleMolecularProfilesUsingPOST"   
#&gt;  [5] "getAllMolecularProfilesUsingGET"                          
#&gt;  [6] "fetchMolecularProfilesUsingPOST"                          
#&gt;  [7] "getMolecularProfileUsingGET"                              
#&gt;  [8] "getDiscreteCopyNumbersInMolecularProfileUsingGET"         
#&gt;  [9] "fetchDiscreteCopyNumbersInMolecularProfileUsingPOST"      
#&gt; [10] "getAllMolecularDataInMolecularProfileUsingGET"            
#&gt; [11] "fetchAllMolecularDataInMolecularProfileUsingPOST"         
#&gt; [12] "getMutationsInMolecularProfileBySampleListIdUsingGET"     
#&gt; [13] "fetchMutationsInMolecularProfileUsingPOST"                
#&gt; [14] "fetchMutationsInMultipleMolecularProfilesUsingPOST"       
#&gt; [15] "getAllMolecularProfilesInStudyUsingGET"                   </div><div class='input'>
<span class='fu'>clinicalData</span>(<span class='kw'>api</span> <span class='kw'>=</span> <span class='no'>cbio</span>, <span class='kw'>studyId</span> <span class='kw'>=</span> <span class='st'>"acc_tcga"</span>)</div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 92 x 70</span><span>
#&gt;    uniquePatientKey patientId studyId AGE   AJCC_PATHOLOGIC… ATYPICAL_MITOTI…
#&gt;    </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>           
#&gt; </span><span style='color: #555555;'> 1</span><span> VENHQS1PUi1BNUo… TCGA-OR-… acc_tc… 23    Stage IV         Atypical Mitoti…
#&gt; </span><span style='color: #555555;'> 2</span><span> VENHQS1PUi1BNUo… TCGA-OR-… acc_tc… 30    Stage III        Atypical Mitoti…
#&gt; </span><span style='color: #555555;'> 3</span><span> VENHQS1PUi1BNUo… TCGA-OR-… acc_tc… 29    Stage II         Atypical Mitoti…
#&gt; </span><span style='color: #555555;'> 4</span><span> VENHQS1PUi1BNUo… TCGA-OR-… acc_tc… 30    Stage III        Atypical Mitoti…
#&gt; </span><span style='color: #555555;'> 5</span><span> VENHQS1PUi1BNUo… TCGA-OR-… acc_tc… 66    Stage III        Atypical Mitoti…
#&gt; </span><span style='color: #555555;'> 6</span><span> VENHQS1PUi1BNUo… TCGA-OR-… acc_tc… 22    Stage II         Atypical Mitoti…
#&gt; </span><span style='color: #555555;'> 7</span><span> VENHQS1PUi1BNUo… TCGA-OR-… acc_tc… 58    Stage II         Atypical Mitoti…
#&gt; </span><span style='color: #555555;'> 8</span><span> VENHQS1PUi1BNUo… TCGA-OR-… acc_tc… 44    Stage IV         Atypical Mitoti…
#&gt; </span><span style='color: #555555;'> 9</span><span> VENHQS1PUi1BNUo… TCGA-OR-… acc_tc… 23    Stage III        Atypical Mitoti…
#&gt; </span><span style='color: #555555;'>10</span><span> VENHQS1PUi1BNUp… TCGA-OR-… acc_tc… 60    Stage II         </span><span style='color: #BB0000;'>NA</span><span>              
#&gt; </span><span style='color: #555555;'># … with 82 more rows, and 64 more variables: CAPSULAR_INVASION </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   CLIN_M_STAGE </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, CLINICAL_STATUS_WITHIN_3_MTHS_SURGERY </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   CT_SCAN_PREOP_RESULTS </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   CYTOPLASM_PRESENCE_LESS_THAN_EQUAL_25_PERCENT </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   DAYS_LAST_FOLLOWUP </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, DAYS_TO_INITIAL_PATHOLOGIC_DIAGNOSIS </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   DFS_MONTHS </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, DFS_STATUS </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, DIFFUSE_ARCHITECTURE </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   ETHNICITY </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, FORM_COMPLETION_DATE </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, HISTOLOGICAL_DIAGNOSIS </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   HISTORY_ADRENAL_HORMONE_EXCESS </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   HISTORY_BASIS_ADRENAL_HORMONE_DX </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, HISTORY_NEOADJUVANT_TRTYN </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   HISTORY_OTHER_MALIGNANCY </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, ICD_10 </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, ICD_O_3_HISTOLOGY </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   ICD_O_3_SITE </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, INFORMED_CONSENT_VERIFIED </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   INITIAL_PATHOLOGIC_DX_YEAR </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, LATERALITY </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   LYMPH_NODE_EXAMINED_COUNT </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, LYMPH_NODES_EXAMINED </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   LYMPH_NODES_EXAMINED_HE_COUNT </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, METASTATIC_DX_CONFIRMED_BY </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   METASTATIC_DX_CONFIRMED_BY_OTHER </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, METASTATIC_SITE_PATIENT </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   MITOSES_PER_50_HPF </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, MITOTIC_RATE </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   MOLECULAR_STUDIES_OTHERS_PERFORMED </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, NECROSIS </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   NEW_TUMOR_EVENT_AFTER_INITIAL_TREATMENT </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, NUCLEAR_GRADE_III_IV </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   OS_MONTHS </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, OS_STATUS </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, OTHER_PATIENT_ID </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   PATH_N_STAGE </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, PATH_T_STAGE </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, PHARM_TX_MITOTANE_ADJUVANT </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   PHARM_TX_MITOTANE_FOR_MACRO_DISEASE </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   PHARM_TX_MITOTANE_INDICATOR </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   PHARM_TX_MITOTANE_THERAPUTIC_AT_PROG </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   PHARM_TX_MITOTANE_THERAPUTIC_AT_REC </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   PHARM_TX_MITOTANE_THERAPUTIC_LEVELS </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   PHARM_TX_MITOTANE_THERAPUTIC_MACRO </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, PHARMACEUTICAL_TX_ADJUVANT </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   PROSPECTIVE_COLLECTION </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, RACE </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   RADIATION_TREATMENT_ADJUVANT </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, RESIDUAL_TUMOR </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, RET </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   RETROSPECTIVE_COLLECTION </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, SAMPLE_COUNT </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, SEX </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   SINUSOID_INVASION </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, SITE_OF_TUMOR_TISSUE </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   TISSUE_SOURCE_SITE </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, TREATMENT_OUTCOME_FIRST_COURSE </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   TUMOR_STATUS </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, VITAL_STATUS </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, WEISS_SCORE_OVERALL </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   WEISS_VENOUS_INVASION </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span></div><div class='input'>
<span class='fu'>molecularProfiles</span>(<span class='kw'>api</span> <span class='kw'>=</span> <span class='no'>cbio</span>, <span class='kw'>studyId</span> <span class='kw'>=</span> <span class='st'>"acc_tcga"</span>)</div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 9 x 7</span><span>
#&gt;   molecularAltera… datatype name  description showProfileInAn… molecularProfil…
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>    </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span>            </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>           
#&gt; </span><span style='color: #555555;'>1</span><span> PROTEIN_LEVEL    LOG2-VA… Prot… Protein ex… FALSE            acc_tcga_rppa   
#&gt; </span><span style='color: #555555;'>2</span><span> PROTEIN_LEVEL    Z-SCORE  Prot… Protein ex… TRUE             acc_tcga_rppa_Z…
#&gt; </span><span style='color: #555555;'>3</span><span> COPY_NUMBER_ALT… DISCRETE Puta… Putative c… TRUE             acc_tcga_gistic 
#&gt; </span><span style='color: #555555;'>4</span><span> MRNA_EXPRESSION  CONTINU… mRNA… mRNA gene … FALSE            acc_tcga_rna_se…
#&gt; </span><span style='color: #555555;'>5</span><span> MRNA_EXPRESSION  Z-SCORE  mRNA… mRNA expre… TRUE             acc_tcga_rna_se…
#&gt; </span><span style='color: #555555;'>6</span><span> COPY_NUMBER_ALT… CONTINU… Rela… Relative l… FALSE            acc_tcga_linear…
#&gt; </span><span style='color: #555555;'>7</span><span> METHYLATION      CONTINU… Meth… Methylatio… FALSE            acc_tcga_methyl…
#&gt; </span><span style='color: #555555;'>8</span><span> MUTATION_EXTEND… MAF      Muta… Mutation d… TRUE             acc_tcga_mutati…
#&gt; </span><span style='color: #555555;'>9</span><span> MRNA_EXPRESSION  Z-SCORE  mRNA… Log-transf… TRUE             acc_tcga_rna_se…
#&gt; </span><span style='color: #555555;'># … with 1 more variable: studyId </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span></div><div class='input'>
<span class='fu'>molecularData</span>(
    <span class='kw'>api</span> <span class='kw'>=</span> <span class='no'>cbio</span>,
    <span class='kw'>molecularProfileId</span> <span class='kw'>=</span> <span class='st'>"acc_tcga_rna_seq_v2_mrna"</span>,
    <span class='kw'>entrezGeneIds</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>1</span>, <span class='fl'>2</span>),
    <span class='kw'>sampleIds</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"TCGA-OR-A5J1-01"</span>, <span class='st'>"TCGA-OR-A5J2-01"</span>)
)</div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 4 x 8</span><span>
#&gt;   uniqueSampleKey uniquePatientKey entrezGeneId molecularProfil… sampleId
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>           </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>                   </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>   
#&gt; </span><span style='color: #555555;'>1</span><span> VENHQS1PUi1BNU… VENHQS1PUi1BNUo…            1 acc_tcga_rna_se… TCGA-OR…
#&gt; </span><span style='color: #555555;'>2</span><span> VENHQS1PUi1BNU… VENHQS1PUi1BNUo…            2 acc_tcga_rna_se… TCGA-OR…
#&gt; </span><span style='color: #555555;'>3</span><span> VENHQS1PUi1BNU… VENHQS1PUi1BNUo…            1 acc_tcga_rna_se… TCGA-OR…
#&gt; </span><span style='color: #555555;'>4</span><span> VENHQS1PUi1BNU… VENHQS1PUi1BNUo…            2 acc_tcga_rna_se… TCGA-OR…
#&gt; </span><span style='color: #555555;'># … with 3 more variables: patientId </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, studyId </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, value </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span></div><div class='input'>
<span class='fu'>sampleLists</span>(<span class='kw'>api</span> <span class='kw'>=</span> <span class='no'>cbio</span>, <span class='kw'>studyId</span> <span class='kw'>=</span> <span class='st'>"acc_tcga"</span>)</div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 9 x 5</span><span>
#&gt;   category          name          description            sampleListId    studyId
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>         </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>                  </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>           </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>  
#&gt; </span><span style='color: #555555;'>1</span><span> all_cases_with_r… Samples with… Samples protein data … acc_tcga_rppa   acc_tc…
#&gt; </span><span style='color: #555555;'>2</span><span> all_cases_with_m… Samples with… Samples with mutation… acc_tcga_cnaseq acc_tc…
#&gt; </span><span style='color: #555555;'>3</span><span> all_cases_in_stu… All samples   All samples (92 sampl… acc_tcga_all    acc_tc…
#&gt; </span><span style='color: #555555;'>4</span><span> all_cases_with_c… Samples with… Samples with CNA data… acc_tcga_cna    acc_tc…
#&gt; </span><span style='color: #555555;'>5</span><span> all_cases_with_m… Samples with… Samples with mutation… acc_tcga_seque… acc_tc…
#&gt; </span><span style='color: #555555;'>6</span><span> all_cases_with_m… Samples with… Samples with methylat… acc_tcga_methy… acc_tc…
#&gt; </span><span style='color: #555555;'>7</span><span> all_cases_with_m… Samples with… Samples with mRNA exp… acc_tcga_rna_s… acc_tc…
#&gt; </span><span style='color: #555555;'>8</span><span> all_cases_with_m… Complete sam… Samples with mutation… acc_tcga_3way_… acc_tc…
#&gt; </span><span style='color: #555555;'>9</span><span> all_cases_with_m… Samples with… Samples with methylat… acc_tcga_methy… acc_tc…</div><div class='input'>
<span class='fu'>samplesInSampleLists</span>(
    <span class='kw'>api</span> <span class='kw'>=</span> <span class='no'>cbio</span>,
    <span class='kw'>sampleListIds</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"acc_tcga_rppa"</span>, <span class='st'>"acc_tcga_cnaseq"</span>)
)</div><div class='output co'>#&gt; CharacterList of length 2
#&gt; [["acc_tcga_cnaseq"]] TCGA-OR-A5J1-01 TCGA-OR-A5J2-01 ... TCGA-PK-A5HC-01
#&gt; [["acc_tcga_rppa"]] TCGA-OR-A5J2-01 TCGA-OR-A5J3-01 ... TCGA-PK-A5HA-01</div><div class='input'>
<span class='fu'>genePanels</span>(<span class='kw'>api</span> <span class='kw'>=</span> <span class='no'>cbio</span>)</div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 46 x 2</span><span>
#&gt;    description                                                       genePanelId
#&gt;    </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>                                                             </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>      
#&gt; </span><span style='color: #555555;'> 1</span><span> Targeted (341 cancer genes) sequencing of various tumor types vi… IMPACT341  
#&gt; </span><span style='color: #555555;'> 2</span><span> Targeted (410 cancer genes) sequencing of various tumor types vi… IMPACT410  
#&gt; </span><span style='color: #555555;'> 3</span><span> Targeted (468 cancer genes) sequencing of various tumor types vi… IMPACT468  
#&gt; </span><span style='color: #555555;'> 4</span><span> Targeted sequencing of urcc tumor via MSK-IMPACT.                 IMPACT     
#&gt; </span><span style='color: #555555;'> 5</span><span> Targeted (300 cancer genes) sequencing of bladder urothelial car… IMPACT300  
#&gt; </span><span style='color: #555555;'> 6</span><span> Targeted deep sequencing (TDS) of 1202 genes on Illumina HiSeq20… UCLA_1202  
#&gt; </span><span style='color: #555555;'> 7</span><span> Targeted sequencing of urcc tumor via MSK-IMPACT.                 IMPACT230  
#&gt; </span><span style='color: #555555;'> 8</span><span> Targeted (27 cancer genes) sequencing of adenoid cystic carcinom… ACYC_FMI_27
#&gt; </span><span style='color: #555555;'> 9</span><span> Targeted (173 cancer genes) sequencing of breast cancers on Illu… METABRIC_1…
#&gt; </span><span style='color: #555555;'>10</span><span> Targeted sequencing of 504 cancer-associated genes on Illumina H… DFCI_504   
#&gt; </span><span style='color: #555555;'># … with 36 more rows</span><span></div><div class='input'>
<span class='fu'>getGenePanel</span>(<span class='kw'>api</span> <span class='kw'>=</span> <span class='no'>cbio</span>, <span class='kw'>genePanelId</span> <span class='kw'>=</span> <span class='st'>"IMPACT341"</span>)</div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 341 x 2</span><span>
#&gt;    entrezGeneId hugoGeneSymbol
#&gt;           </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>         
#&gt; </span><span style='color: #555555;'> 1</span><span>           25 ABL1          
#&gt; </span><span style='color: #555555;'> 2</span><span>        </span><span style='text-decoration: underline;'>84</span><span>142 ABRAXAS1      
#&gt; </span><span style='color: #555555;'> 3</span><span>          207 AKT1          
#&gt; </span><span style='color: #555555;'> 4</span><span>          208 AKT2          
#&gt; </span><span style='color: #555555;'> 5</span><span>        </span><span style='text-decoration: underline;'>10</span><span>000 AKT3          
#&gt; </span><span style='color: #555555;'> 6</span><span>          238 ALK           
#&gt; </span><span style='color: #555555;'> 7</span><span>          242 ALOX12B       
#&gt; </span><span style='color: #555555;'> 8</span><span>       </span><span style='text-decoration: underline;'>139</span><span>285 AMER1         
#&gt; </span><span style='color: #555555;'> 9</span><span>          324 APC           
#&gt; </span><span style='color: #555555;'>10</span><span>          367 AR            
#&gt; </span><span style='color: #555555;'># … with 331 more rows</span><span></div><div class='input'>

<span class='fu'>getDataByGenePanel</span>(<span class='no'>cbio</span>, <span class='kw'>studyId</span> <span class='kw'>=</span> <span class='st'>"acc_tcga"</span>, <span class='kw'>genePanelId</span> <span class='kw'>=</span> <span class='st'>"IMPACT341"</span>,
   <span class='kw'>molecularProfileId</span> <span class='kw'>=</span> <span class='st'>"acc_tcga_rppa"</span>)</div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 2,622 x 8</span><span>
#&gt;    uniqueSampleKey uniquePatientKey entrezGeneId molecularProfil… sampleId
#&gt;    </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>           </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>                   </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>   
#&gt; </span><span style='color: #555555;'> 1</span><span> VENHQS1PUi1BNU… VENHQS1PUi1BNUo…          207 acc_tcga_rppa    TCGA-OR…
#&gt; </span><span style='color: #555555;'> 2</span><span> VENHQS1PUi1BNU… VENHQS1PUi1BNUo…          208 acc_tcga_rppa    TCGA-OR…
#&gt; </span><span style='color: #555555;'> 3</span><span> VENHQS1PUi1BNU… VENHQS1PUi1BNUo…        </span><span style='text-decoration: underline;'>10</span><span>000 acc_tcga_rppa    TCGA-OR…
#&gt; </span><span style='color: #555555;'> 4</span><span> VENHQS1PUi1BNU… VENHQS1PUi1BNUo…          367 acc_tcga_rppa    TCGA-OR…
#&gt; </span><span style='color: #555555;'> 5</span><span> VENHQS1PUi1BNU… VENHQS1PUi1BNUo…          472 acc_tcga_rppa    TCGA-OR…
#&gt; </span><span style='color: #555555;'> 6</span><span> VENHQS1PUi1BNU… VENHQS1PUi1BNUo…         </span><span style='text-decoration: underline;'>8</span><span>314 acc_tcga_rppa    TCGA-OR…
#&gt; </span><span style='color: #555555;'> 7</span><span> VENHQS1PUi1BNU… VENHQS1PUi1BNUo…          596 acc_tcga_rppa    TCGA-OR…
#&gt; </span><span style='color: #555555;'> 8</span><span> VENHQS1PUi1BNU… VENHQS1PUi1BNUo…          598 acc_tcga_rppa    TCGA-OR…
#&gt; </span><span style='color: #555555;'> 9</span><span> VENHQS1PUi1BNU… VENHQS1PUi1BNUo…        </span><span style='text-decoration: underline;'>10</span><span>018 acc_tcga_rppa    TCGA-OR…
#&gt; </span><span style='color: #555555;'>10</span><span> VENHQS1PUi1BNU… VENHQS1PUi1BNUo…          673 acc_tcga_rppa    TCGA-OR…
#&gt; </span><span style='color: #555555;'># … with 2,612 more rows, and 3 more variables: patientId </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, studyId </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   value </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span></div><div class='input'>
</div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Levi Waldron, Marcel Ramos.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


