<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Manually download, untar, and load study tarballs — downloadStudy • cBioPortalData</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Manually download, untar, and load study tarballs — downloadStudy"><meta property="og:description" content="Note that these functions should be used when a particular
study is not currently available as a MultiAssayExperiment
representation. Otherwise, use cBioDataPack. Provide a cancer_study_id
from getStudies and retrieve the study tarball from the cBio
Genomics Portal.  These functions are used by cBioDataPack under the hood
to download,untar, and load the tarball datasets with caching. As stated in
?cBioDataPack, not all studies are currently working as
MultiAssayExperiment objects. As of July 2020, about ~80% of
datasets can be successfully imported into the MultiAssayExperiment data
class. Please open an issue if you would like the team to prioritize a
study. You may also check getStudies(buildReport = TRUE)$pack_build
for the current status."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cBioPortalData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.10.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/cBioPortalData.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/cBioPortalDataErrors.html">cBioPortalData: Data Build Errors</a>
    </li>
    <li>
      <a href="../articles/cBioPortalRClient.html">cBioPortalData: API Reference Guide for Devs</a>
    </li>
    <li>
      <a href="../articles/cgdsrMigration.html">cgdsr to cBioPortalData: Migration Tutorial</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/waldronlab/cBioPortalData/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Manually download, untar, and load study tarballs</h1>
    <small class="dont-index">Source: <a href="https://github.com/waldronlab/cBioPortalData/blob/HEAD/R/cBioDataPack.R" class="external-link"><code>R/cBioDataPack.R</code></a></small>
    <div class="hidden name"><code>downloadStudy.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><strong>Note</strong> that these functions should be used when a particular
study is <em>not</em> currently available as a <code>MultiAssayExperiment</code>
representation. Otherwise, use <code>cBioDataPack</code>. Provide a <code>cancer_study_id</code>
from <code>getStudies</code> and retrieve the study tarball from the cBio
Genomics Portal.  These functions are used by <code>cBioDataPack</code> under the hood
to download,untar, and load the tarball datasets with caching. As stated in
<code><a href="cBioDataPack.html">?cBioDataPack</a></code>, not all studies are currently working as
<code>MultiAssayExperiment</code> objects. As of July 2020, about ~80% of
datasets can be successfully imported into the <code>MultiAssayExperiment</code> data
class. Please open an issue if you would like the team to prioritize a
study. You may also check <code>getStudies(buildReport = TRUE)$pack_build</code>
for the current status.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">downloadStudy</span><span class="op">(</span></span>
<span>  <span class="va">cancer_study_id</span>,</span>
<span>  use_cache <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  force <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  url_location <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"cBio_URL"</span>, <span class="va">.url_location</span><span class="op">)</span>,</span>
<span>  ask <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">untarStudy</span><span class="op">(</span><span class="va">cancer_study_file</span>, exdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">loadStudy</span><span class="op">(</span></span>
<span>  <span class="va">filepath</span>,</span>
<span>  names.field <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Hugo_Symbol"</span>, <span class="st">"Entrez_Gene_Id"</span>, <span class="st">"Gene"</span><span class="op">)</span>,</span>
<span>  cleanup <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>cancer_study_id</dt>
<dd><p>character(1) The study identifier from cBioPortal as
in <a href="https://cbioportal.org/webAPI" class="external-link">https://cbioportal.org/webAPI</a></p></dd>


<dt>use_cache</dt>
<dd><p>logical(1) (default TRUE) create the default cache location
and use it to track downloaded data. If data found in the cache, data will
not be re-downloaded. A path can also be provided to data cache location.</p></dd>


<dt>force</dt>
<dd><p>logical(1) (default FALSE) whether to force re-download data
from remote location</p></dd>


<dt>url_location</dt>
<dd><p>character(1)
(default "https://cbioportal-datahub.s3.amazonaws.com") the URL location for
downloading packaged data. Can be set using the 'cBio_URL' option (see
<code><a href="cBioDataPack.html">?cBioDataPack</a></code> for more details)</p></dd>


<dt>ask</dt>
<dd><p>logical(1) Whether to prompt the the user before downloading and
loading study <code>MultiAssayExperiment</code> that is not currently building based
on previous testing. Set to <code><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive()</a></code> by default. In a
non-interactive session, data download will be attempted; equivalent to
<code>ask = FALSE</code>. The argument will also be used when a cache directory needs
to be created when using <code>downloadStudy</code>.</p></dd>


<dt>cancer_study_file</dt>
<dd><p>character(1) indicates the on-disk location
of the downloaded tarball</p></dd>


<dt>exdir</dt>
<dd><p>character(1) indicates the folder location to <em>put</em>
the contents of the tarball (default <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code>; see also <code><a href="https://rdrr.io/r/utils/untar.html" class="external-link">?untar</a></code>)</p></dd>


<dt>filepath</dt>
<dd><p>character(1) indicates the folder location where
the contents of the tarball are <em>located</em> (usually the same as <code>exdir</code>)</p></dd>


<dt>names.field</dt>
<dd><p>character() Possible column names for the
column that will used to label ranges for data such as mutations or copy
number (default: <code>c("Hugo_Symbol", "Entrez_Gene_Id", "Gene")</code>). Values are
cycled through and eliminated when no data present, or duplicates are found.
Values in the corresponding column must be unique in each row.</p></dd>


<dt>cleanup</dt>
<dd><p>logical(1) whether to delete the <code>untar</code>-red contents from
the <code>exdir</code> folder (default TRUE)</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<ul><li><p>downloadStudy - The file location of the data tarball</p></li>
<li><p>untarStudy - The directory location of the contents</p></li>
<li><p>loadStudy - A MultiAssayExperiment-class object</p></li>
</ul></div>
    <div id="details">
    <h2>Details</h2>
    <p>When attempting to load a dataset using <code>loadStudy</code>, note that
the <code>cleanup</code> argument is set to <code>TRUE</code> by default. Change the argument
to <code>FALSE</code> if you would like to keep the untarred data in the <code>exdir</code>
location. <code>downloadStudy</code> and <code>untarStudy</code> are not affected by this change.
The tarball of the downloaded data is cached via <code>BiocFileCache</code> when
<code>use_cache</code> is <code>TRUE</code>.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><a href="cBioDataPack.html">cBioDataPack</a>, <a href="http://waldronlab.io/MultiAssayExperiment/reference/MultiAssayExperiment-class.html" class="external-link">MultiAssayExperiment</a></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="op">(</span><span class="va">acc_file</span> <span class="op">&lt;-</span> <span class="fu">downloadStudy</span><span class="op">(</span><span class="st">"acc_tcga"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Study file in cache: acc_tcga</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                              BFC16 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "/github/home/.cache/R/cBioPortalData/d0c1b40f5b1_acc_tcga.tar.gz" </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="op">(</span><span class="va">file_dir</span> <span class="op">&lt;-</span> <span class="fu">untarStudy</span><span class="op">(</span><span class="va">acc_file</span>, <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/tmp/Rtmp3USpIe/d0c1b40f5b1_acc_tcga"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">loadStudy</span><span class="op">(</span><span class="va">file_dir</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Working on: /tmp/Rtmp3USpIe/d0c1b40f5b1_acc_tcga/acc_tcga/data_cna.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Working on: /tmp/Rtmp3USpIe/d0c1b40f5b1_acc_tcga/acc_tcga/data_cna_hg19.seg</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Working on: /tmp/Rtmp3USpIe/d0c1b40f5b1_acc_tcga/acc_tcga/data_linear_cna.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Working on: /tmp/Rtmp3USpIe/d0c1b40f5b1_acc_tcga/acc_tcga/data_methylation_hm450.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Working on: /tmp/Rtmp3USpIe/d0c1b40f5b1_acc_tcga/acc_tcga/data_mrna_seq_v2_rsem.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Working on: /tmp/Rtmp3USpIe/d0c1b40f5b1_acc_tcga/acc_tcga/data_mrna_seq_v2_rsem_zscores_ref_all_samples.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Working on: /tmp/Rtmp3USpIe/d0c1b40f5b1_acc_tcga/acc_tcga/data_mrna_seq_v2_rsem_zscores_ref_diploid_samples.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Working on: /tmp/Rtmp3USpIe/d0c1b40f5b1_acc_tcga/acc_tcga/data_mutations.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Working on: /tmp/Rtmp3USpIe/d0c1b40f5b1_acc_tcga/acc_tcga/data_mutsig.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Working on: /tmp/Rtmp3USpIe/d0c1b40f5b1_acc_tcga/acc_tcga/data_rppa.txt</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Working on: /tmp/Rtmp3USpIe/d0c1b40f5b1_acc_tcga/acc_tcga/data_rppa_zscores.txt</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span> Multiple prefixes found, using keyword 'region' or taking first one</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span> Multiple prefixes found, using keyword 'region' or taking first one</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A MultiAssayExperiment object of 10 listed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments with user-defined names and respective classes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Containing an ExperimentList class object of length 10:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] cna: SummarizedExperiment with 24776 rows and 90 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [2] cna_hg19.seg: RaggedExperiment with 16080 rows and 90 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] linear_cna: SummarizedExperiment with 24776 rows and 90 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] methylation_hm450: SummarizedExperiment with 15754 rows and 80 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [5] mrna_seq_v2_rsem: SummarizedExperiment with 20531 rows and 79 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [6] mrna_seq_v2_rsem_zscores_ref_all_samples: SummarizedExperiment with 20531 rows and 79 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7] mrna_seq_v2_rsem_zscores_ref_diploid_samples: SummarizedExperiment with 20440 rows and 79 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [8] mutations: RaggedExperiment with 20166 rows and 90 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [9] rppa: SummarizedExperiment with 192 rows and 46 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [10] rppa_zscores: SummarizedExperiment with 191 rows and 46 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Functionality:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments() - obtain the ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  colData() - the primary/phenotype DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  sampleMap() - the sample coordination DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  *Format() - convert into a long or wide DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  exportClass() - save data to flat files</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Levi Waldron, Marcel Ramos.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

