<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>cBioPortalData: User Guide • cBioPortalData</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="cBioPortalData: User Guide">
<meta property="og:description" content="cBioPortalData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cBioPortalData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.10.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/cBioPortalData.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cBioPortalDataErrors.html">cBioPortalData: Data Build Errors</a>
    </li>
    <li>
      <a href="../articles/cBioPortalRClient.html">cBioPortalData: API Reference Guide for Devs</a>
    </li>
    <li>
      <a href="../articles/cgdsrMigration.html">cgdsr to cBioPortalData: Migration Tutorial</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/cBioPortalData/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>cBioPortalData: User Guide</h1>
                        <h4 data-toc-skip class="author">Marcel Ramos
&amp; Levi Waldron</h4>
            
            <h4 data-toc-skip class="date">January 03, 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/cBioPortalData/blob/HEAD/vignettes/cBioPortalData.Rmd" class="external-link"><code>vignettes/cBioPortalData.Rmd</code></a></small>
      <div class="hidden name"><code>cBioPortalData.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">cBioPortalData</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">AnVIL</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The cBioPortal for Cancer Genomics <a href="https://cbioportal.org" class="external-link">website</a> is a great resource for
interactive exploration of study datasets. However, it does not easily
allow the analyst to obtain and further analyze the data.</p>
<p>We’ve developed the <code>cBioPortalData</code> package to fill this
need to programmatically access the data resources available on the
cBioPortal.</p>
<p>The <code>cBioPortalData</code> package provides an R interface for
accessing the cBioPortal study data within the Bioconductor
ecosystem.</p>
<p>It downloads study data from the cBioPortal API (the full API
specification can be found here <a href="https://cbioportal.org/api" class="external-link uri">https://cbioportal.org/api</a>) and uses Bioconductor
infrastructure to cache and represent the data.</p>
<p>We use the <a href="https://dx.doi.org/10.1158/0008-5472.CAN-17-0344" class="external-link"><code>MultiAssayExperiment</code></a>
(<span class="citation">@Ramos2017-er</span>) package to integrate,
represent, and coordinate multiple experiments for the studies availble
in the cBioPortal. This package in conjunction with
<code>curatedTCGAData</code> give access to a large trove of publicly
available bioinformatic data. Please see our publication <a href="https://dx.doi.org/10.1200/CCI.19.00119" class="external-link">here</a> (<span class="citation">@Ramos2020-ya</span>).</p>
<p>We demonstrate common use cases of <code>cBioPortalData</code> and
<code>curatedTCGAData</code> during Bioconductor conference <a href="https://waldronlab.io/MultiAssayWorkshop/" class="external-link">workshops</a>.</p>
<div class="section level3">
<h3 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h3>
<div class="section level4">
<h4 id="data-structures">Data Structures<a class="anchor" aria-label="anchor" href="#data-structures"></a>
</h4>
<p>Data are provided as a single <code>MultiAssayExperiment</code> per
study. The <code>MultiAssayExperiment</code> representation usually
contains <code>SummarizedExperiment</code> objects for expression data
and <code>RaggedExperiment</code> objects for mutation and CNV-type
data. <code>RaggedExperiment</code> is a data class for representing
‘ragged’ genomic location data, meaning that the measurements per sample
vary.</p>
<p>For more information, please see the <code>RaggedExperiment</code>
and <code>SummarizedExperiment</code> vignettes.</p>
</div>
<div class="section level4">
<h4 id="identifying-available-studies">Identifying available studies<a class="anchor" aria-label="anchor" href="#identifying-available-studies"></a>
</h4>
<p>As we work through the data, there are some datasest that cannot be
represented as <code>MultiAssayExperiment</code> objects. This can be
due to a number of reasons such as the way the data is handled, presence
of mis-matched identifiers, invalid data types, etc. To see what
datasets are currently not building, we can look refer to
<code><a href="../reference/cBioPortal.html">getStudies()</a></code> with the <code>buildReport = TRUE</code>
argument.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cbio</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortal.html">cBioPortal</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in .service_validate_md5sum(api_reference_url, api_reference_md5sum, : service version differs from validated version</span></span>
<span><span class="co">##     service url: https://www.cbioportal.org/api/v2/api-docs</span></span>
<span><span class="co">##     observed md5sum: 07ceb76cc5afcf54a9cf2e1a689b18f7</span></span>
<span><span class="co">##     expected md5sum: 4188a76d21595ffbf4d7749d60f24898</span></span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">studies</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortal.html">getStudies</a></span><span class="op">(</span><span class="va">cbio</span>, buildReport <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">studies</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 15</span></span></span>
<span><span class="co">##   name     descr…¹ publi…² groups status impor…³ allSa…⁴ readP…⁵ studyId cance…⁶</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> Adrenoc… <span style="color: #949494;">"</span>TCGA … TRUE    <span style="color: #949494;">"</span>PUBL…      0 2022-1…      92 TRUE    acc_tc… acc    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> Acute L… <span style="color: #949494;">"</span>Compr… TRUE    <span style="color: #949494;">"</span>PUBL…      0 2022-1…      93 TRUE    all_st… bll    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> Hypodip… <span style="color: #949494;">"</span>Whole… TRUE    <span style="color: #949494;">""</span>          0 2022-1…      44 TRUE    all_st… myeloid</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> Adenoid… <span style="color: #949494;">"</span>Whole… TRUE    <span style="color: #949494;">"</span>ACYC…      0 2022-1…      12 TRUE    acbc_m… acbc   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> Adenoid… <span style="color: #949494;">"</span>Targe… TRUE    <span style="color: #949494;">"</span>ACYC…      0 2022-1…      28 TRUE    acyc_f… acyc   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> Adenoid… <span style="color: #949494;">"</span>Whole… TRUE    <span style="color: #949494;">"</span>ACYC…      0 2022-1…      25 TRUE    acyc_j… acyc   </span></span>
<span><span class="co">## <span style="color: #949494;"># … with 5 more variables: referenceGenome &lt;chr&gt;, pmid &lt;chr&gt;, citation &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   api_build &lt;lgl&gt;, pack_build &lt;lgl&gt;, and abbreviated variable names</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ¹​description, ²​publicStudy, ³​importDate, ⁴​allSampleCount, ⁵​readPermission,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ⁶​cancerTypeId</span></span></span></code></pre>
<p>The last two columns will show the availability of each
<code>studyId</code> for either download method (<code>pack_build</code>
for <code>cBioDataPack</code> and <code>api_build</code> for
<code>cBioPortalData</code>).</p>
</div>
<div class="section level4">
<h4 id="choosing-download-method">Choosing download method<a class="anchor" aria-label="anchor" href="#choosing-download-method"></a>
</h4>
<p>There are two main user-facing functions for downloading data from
the cBioPortal API.</p>
<ul>
<li><p><code>cBioDataPack</code> makes use of the tarball distribution
of study data. This is useful when the user wants to download and
analyze the entirety of the data as available from the cBioPortal.org
website.</p></li>
<li><p><code>cBioPortalData</code> allows a more flexibile approach to
obtaining study data based on the available parameters such as molecular
profile identifiers. This option is useful for users who have a set of
gene symbols or identifiers and would like to get a smaller subset of
the data that correspond to a particular molecular profile.</p></li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="two-main-functions">Two main functions<a class="anchor" aria-label="anchor" href="#two-main-functions"></a>
</h3>
<div class="section level4">
<h4 id="cbiodatapack-obtain-study-data-as-zipped-tarballs">cBioDataPack: Obtain Study Data as Zipped Tarballs<a class="anchor" aria-label="anchor" href="#cbiodatapack-obtain-study-data-as-zipped-tarballs"></a>
</h4>
<p>This function will access the packaged data from and return an
integrative MultiAssayExperiment representation.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Use ask=FALSE for non-interactive use</span></span>
<span><span class="va">laml</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioDataPack.html">cBioDataPack</a></span><span class="op">(</span><span class="st">"laml_tcga"</span>, ask <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">laml</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 12 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 12:</span></span>
<span><span class="co">##  [1] cna_hg19.seg: RaggedExperiment with 13571 rows and 191 columns</span></span>
<span><span class="co">##  [2] cna: SummarizedExperiment with 24776 rows and 191 columns</span></span>
<span><span class="co">##  [3] linear_cna: SummarizedExperiment with 24776 rows and 191 columns</span></span>
<span><span class="co">##  [4] methylation_hm27: SummarizedExperiment with 10968 rows and 194 columns</span></span>
<span><span class="co">##  [5] methylation_hm450: SummarizedExperiment with 10968 rows and 194 columns</span></span>
<span><span class="co">##  [6] mrna_seq_rpkm_zscores_ref_all_samples: SummarizedExperiment with 19720 rows and 179 columns</span></span>
<span><span class="co">##  [7] mrna_seq_rpkm_zscores_ref_diploid_samples: SummarizedExperiment with 19719 rows and 179 columns</span></span>
<span><span class="co">##  [8] mrna_seq_rpkm: SummarizedExperiment with 19720 rows and 179 columns</span></span>
<span><span class="co">##  [9] mrna_seq_v2_rsem_zscores_ref_all_samples: SummarizedExperiment with 20531 rows and 173 columns</span></span>
<span><span class="co">##  [10] mrna_seq_v2_rsem_zscores_ref_diploid_samples: SummarizedExperiment with 20440 rows and 173 columns</span></span>
<span><span class="co">##  [11] mrna_seq_v2_rsem: SummarizedExperiment with 20531 rows and 173 columns</span></span>
<span><span class="co">##  [12] mutations: RaggedExperiment with 2584 rows and 197 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="cbioportaldata-obtain-data-from-the-cbioportal-api">cBioPortalData: Obtain data from the cBioPortal API<a class="anchor" aria-label="anchor" href="#cbioportaldata-obtain-data-from-the-cbioportal-api"></a>
</h4>
<p>This function provides a more flexible and granular way to request a
MultiAssayExperiment object from a study ID, molecular profile, gene
panel, sample list.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">acc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortalData.html">cBioPortalData</a></span><span class="op">(</span>api <span class="op">=</span> <span class="va">cbio</span>, by <span class="op">=</span> <span class="st">"hugoGeneSymbol"</span>, studyId <span class="op">=</span> <span class="st">"acc_tcga"</span>,</span>
<span>    genePanelId <span class="op">=</span> <span class="st">"IMPACT341"</span>,</span>
<span>    molecularProfileIds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"acc_tcga_rppa"</span>, <span class="st">"acc_tcga_linear_CNA"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 1 colData rownames not in sampleMap 'primary'</span></span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">acc</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 2 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 2:</span></span>
<span><span class="co">##  [1] acc_tcga_linear_CNA: SummarizedExperiment with 339 rows and 90 columns</span></span>
<span><span class="co">##  [2] acc_tcga_rppa: SummarizedExperiment with 57 rows and 46 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<p>Note. To avoid overloading the API service, the API was designed to
only query a part of the study data. Therefore, the user is required to
enter either a set of genes of interest or a gene panel identifier.</p>
</div>
</div>
<div class="section level3">
<h3 id="clearing-the-cache">Clearing the cache<a class="anchor" aria-label="anchor" href="#clearing-the-cache"></a>
</h3>
<div class="section level4">
<h4 id="cbiodatapack">cBioDataPack<a class="anchor" aria-label="anchor" href="#cbiodatapack"></a>
</h4>
<p>In cases where a download is interrupted, the user may experience a
corrupt cache. The user can clear the cache for a particular study by
using the <code>removeCache</code> function. Note that this function
only works for data downloaded through the <code>cBioDataPack</code>
function.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">removeCache</span><span class="op">(</span><span class="st">"laml_tcga"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="cbioportaldata">cBioPortalData<a class="anchor" aria-label="anchor" href="#cbioportaldata"></a>
</h4>
<p>For users who wish to clear the entire <code>cBioPortalData</code>
cache, it is recommended that they use:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="st">"~/.cache/cBioPortalData/"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="example-analysis-kaplan-meier-plot">Example Analysis: Kaplan-Meier Plot<a class="anchor" aria-label="anchor" href="#example-analysis-kaplan-meier-plot"></a>
</h3>
<p>We can use information in the <code>colData</code> to draw a K-M plot
with a few variables from the <code>colData</code> slot of the
<code>MultiAssayExperiment</code>. First, we load the necessary
packages:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/therneau/survival" class="external-link">survival</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/survminer/index.html" class="external-link">survminer</a></span><span class="op">)</span></span></code></pre></div>
<p>We can check the data to lookout for any issues.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">laml</span><span class="op">)</span><span class="op">$</span><span class="va">OS_STATUS</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##   0:LIVING 1:DECEASED </span></span>
<span><span class="co">##         67        133</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">laml</span><span class="op">)</span><span class="op">$</span><span class="va">OS_MONTHS</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "character"</span></span></code></pre>
<p>Now, we clean the data a bit to ensure that our variables are of the
right type for the subsequent survival model fit.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">collaml</span> <span class="op">&lt;-</span> <span class="fu">colData</span><span class="op">(</span><span class="va">laml</span><span class="op">)</span></span>
<span><span class="va">collaml</span><span class="op">[</span><span class="va">collaml</span><span class="op">$</span><span class="va">OS_MONTHS</span> <span class="op">==</span> <span class="st">"[Not Available]"</span>, <span class="st">"OS_MONTHS"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span>
<span><span class="va">collaml</span><span class="op">$</span><span class="va">OS_MONTHS</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">collaml</span><span class="op">$</span><span class="va">OS_MONTHS</span><span class="op">)</span></span>
<span><span class="fu">colData</span><span class="op">(</span><span class="va">laml</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">collaml</span></span></code></pre></div>
<p>We specify a simple survival model using <code>SEX</code> as a
covariate and we draw the K-M plot.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survival/man/survfit.html" class="external-link">survfit</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/survival/man/Surv.html" class="external-link">Surv</a></span><span class="op">(</span><span class="va">OS_MONTHS</span>, <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substr</a></span><span class="op">(</span><span class="va">OS_STATUS</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">~</span> <span class="va">SEX</span>,</span>
<span>    data <span class="op">=</span> <span class="fu">colData</span><span class="op">(</span><span class="va">laml</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/survminer/man/ggsurvplot.html" class="external-link">ggsurvplot</a></span><span class="op">(</span><span class="va">fit</span>, data <span class="op">=</span> <span class="fu">colData</span><span class="op">(</span><span class="va">laml</span><span class="op">)</span>, risk.table <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="cBioPortalData_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<div class="section level4">
<h4 id="data-update-requests">Data update requests<a class="anchor" aria-label="anchor" href="#data-update-requests"></a>
</h4>
<p>If you are interested in a particular study dataset that is not
currently building, please open an issue at our GitHub repository <a href="https://github.com/waldronlab/cBioPortalData/issues" class="external-link">location</a>
and we will do our best to resolve the issues with either the data or
the code.</p>
<p>We appreciate your feedback!</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="sessioninfo">sessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.2 (2022-10-31)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.1 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] survminer_0.4.9             ggpubr_0.5.0               </span></span>
<span><span class="co">##  [3] ggplot2_3.4.0               survival_3.4-0             </span></span>
<span><span class="co">##  [5] cBioPortalData_2.10.3       MultiAssayExperiment_1.24.0</span></span>
<span><span class="co">##  [7] SummarizedExperiment_1.28.0 Biobase_2.58.0             </span></span>
<span><span class="co">##  [9] GenomicRanges_1.50.2        GenomeInfoDb_1.34.4        </span></span>
<span><span class="co">## [11] IRanges_2.32.0              S4Vectors_0.36.1           </span></span>
<span><span class="co">## [13] BiocGenerics_0.44.0         MatrixGenerics_1.10.0      </span></span>
<span><span class="co">## [15] matrixStats_0.63.0          AnVIL_1.10.1               </span></span>
<span><span class="co">## [17] dplyr_1.0.10                BiocStyle_2.26.0           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] backports_1.4.1           BiocBaseUtils_1.0.0      </span></span>
<span><span class="co">##   [3] BiocFileCache_2.6.0       systemfonts_1.0.4        </span></span>
<span><span class="co">##   [5] RCircos_1.2.2             splines_4.2.2            </span></span>
<span><span class="co">##   [7] BiocParallel_1.32.5       TCGAutils_1.18.0         </span></span>
<span><span class="co">##   [9] digest_0.6.31             htmltools_0.5.4          </span></span>
<span><span class="co">##  [11] fansi_1.0.3               magrittr_2.0.3           </span></span>
<span><span class="co">##  [13] memoise_2.0.1             tzdb_0.3.0               </span></span>
<span><span class="co">##  [15] limma_3.54.0              Biostrings_2.66.0        </span></span>
<span><span class="co">##  [17] readr_2.1.3               vroom_1.6.0              </span></span>
<span><span class="co">##  [19] pkgdown_2.0.7             prettyunits_1.1.1        </span></span>
<span><span class="co">##  [21] colorspace_2.0-3          blob_1.2.3               </span></span>
<span><span class="co">##  [23] rvest_1.0.3               rappdirs_0.3.3           </span></span>
<span><span class="co">##  [25] textshaping_0.3.6         xfun_0.36                </span></span>
<span><span class="co">##  [27] crayon_1.5.2              RCurl_1.98-1.9           </span></span>
<span><span class="co">##  [29] jsonlite_1.8.4            RaggedExperiment_1.22.0  </span></span>
<span><span class="co">##  [31] zoo_1.8-11                glue_1.6.2               </span></span>
<span><span class="co">##  [33] GenomicDataCommons_1.22.0 gtable_0.3.1             </span></span>
<span><span class="co">##  [35] zlibbioc_1.44.0           XVector_0.38.0           </span></span>
<span><span class="co">##  [37] DelayedArray_0.24.0       car_3.1-1                </span></span>
<span><span class="co">##  [39] abind_1.4-5               scales_1.2.1             </span></span>
<span><span class="co">##  [41] futile.options_1.0.1      DBI_1.1.3                </span></span>
<span><span class="co">##  [43] rstatix_0.7.1             miniUI_0.1.1.1           </span></span>
<span><span class="co">##  [45] Rcpp_1.0.9                gridtext_0.1.5           </span></span>
<span><span class="co">##  [47] xtable_1.8-4              progress_1.2.2           </span></span>
<span><span class="co">##  [49] bit_4.0.5                 km.ci_0.5-6              </span></span>
<span><span class="co">##  [51] DT_0.26                   htmlwidgets_1.6.0        </span></span>
<span><span class="co">##  [53] httr_1.4.4                ellipsis_0.3.2           </span></span>
<span><span class="co">##  [55] farver_2.1.1              pkgconfig_2.0.3          </span></span>
<span><span class="co">##  [57] XML_3.99-0.13             rapiclient_0.1.3         </span></span>
<span><span class="co">##  [59] sass_0.4.4                dbplyr_2.2.1             </span></span>
<span><span class="co">##  [61] utf8_1.2.2                RJSONIO_1.3-1.6          </span></span>
<span><span class="co">##  [63] labeling_0.4.2            tidyselect_1.2.0         </span></span>
<span><span class="co">##  [65] rlang_1.0.6               later_1.3.0              </span></span>
<span><span class="co">##  [67] AnnotationDbi_1.60.0      munsell_0.5.0            </span></span>
<span><span class="co">##  [69] tools_4.2.2               cachem_1.0.6             </span></span>
<span><span class="co">##  [71] cli_3.5.0                 generics_0.1.3           </span></span>
<span><span class="co">##  [73] RSQLite_2.2.20            broom_1.0.2              </span></span>
<span><span class="co">##  [75] evaluate_0.19             stringr_1.5.0            </span></span>
<span><span class="co">##  [77] fastmap_1.1.0             yaml_2.3.6               </span></span>
<span><span class="co">##  [79] ragg_1.2.4                knitr_1.41               </span></span>
<span><span class="co">##  [81] bit64_4.0.5               fs_1.5.2                 </span></span>
<span><span class="co">##  [83] survMisc_0.5.6            purrr_1.0.0              </span></span>
<span><span class="co">##  [85] KEGGREST_1.38.0           mime_0.12                </span></span>
<span><span class="co">##  [87] formatR_1.13              xml2_1.3.3               </span></span>
<span><span class="co">##  [89] biomaRt_2.54.0            compiler_4.2.2           </span></span>
<span><span class="co">##  [91] filelock_1.0.2            curl_4.3.3               </span></span>
<span><span class="co">##  [93] png_0.1-8                 ggsignif_0.6.4           </span></span>
<span><span class="co">##  [95] tibble_3.1.8              bslib_0.4.2              </span></span>
<span><span class="co">##  [97] stringi_1.7.8             highr_0.10               </span></span>
<span><span class="co">##  [99] futile.logger_1.4.3       GenomicFeatures_1.50.3   </span></span>
<span><span class="co">## [101] desc_1.4.2                lattice_0.20-45          </span></span>
<span><span class="co">## [103] Matrix_1.5-3              commonmark_1.8.1         </span></span>
<span><span class="co">## [105] markdown_1.4              KMsurv_0.1-5             </span></span>
<span><span class="co">## [107] RTCGAToolbox_2.28.0       vctrs_0.5.1              </span></span>
<span><span class="co">## [109] pillar_1.8.1              lifecycle_1.0.3          </span></span>
<span><span class="co">## [111] BiocManager_1.30.19.5     jquerylib_0.1.4          </span></span>
<span><span class="co">## [113] data.table_1.14.6         bitops_1.0-7             </span></span>
<span><span class="co">## [115] httpuv_1.6.7              rtracklayer_1.58.0       </span></span>
<span><span class="co">## [117] R6_2.5.1                  BiocIO_1.8.0             </span></span>
<span><span class="co">## [119] bookdown_0.31             promises_1.2.0.1         </span></span>
<span><span class="co">## [121] gridExtra_2.3             codetools_0.2-18         </span></span>
<span><span class="co">## [123] lambda.r_1.2.4            assertthat_0.2.1         </span></span>
<span><span class="co">## [125] rprojroot_2.0.3           rjson_0.2.21             </span></span>
<span><span class="co">## [127] withr_2.5.0               GenomicAlignments_1.34.0 </span></span>
<span><span class="co">## [129] Rsamtools_2.14.0          GenomeInfoDbData_1.2.9   </span></span>
<span><span class="co">## [131] ggtext_0.1.2              parallel_4.2.2           </span></span>
<span><span class="co">## [133] hms_1.1.2                 grid_4.2.2               </span></span>
<span><span class="co">## [135] tidyr_1.2.1               rmarkdown_2.19           </span></span>
<span><span class="co">## [137] carData_3.0-5             shiny_1.7.4              </span></span>
<span><span class="co">## [139] restfulr_0.0.15</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Levi Waldron, Marcel Ramos.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
