<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>cBioPortalData: API Reference Guide for Devs • cBioPortalData</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="cBioPortalData: API Reference Guide for Devs">
<meta property="og:description" content="cBioPortalData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cBioPortalData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.14.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/cBioPortalData.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cBioPortalDataErrors.html">cBioPortalData: Data Build Errors</a>
    </li>
    <li>
      <a href="../articles/cBioPortalRClient.html">cBioPortalData: API Reference Guide for Devs</a>
    </li>
    <li>
      <a href="../articles/cgdsrMigration.html">cgdsr to cBioPortalData: Migration Tutorial</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/cBioPortalData/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>cBioPortalData: API Reference Guide for
Devs</h1>
                        <h4 data-toc-skip class="author">Marcel Ramos
&amp; Levi Waldron</h4>
            
            <h4 data-toc-skip class="date">February 02, 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/cBioPortalData/blob/HEAD/vignettes/cBioPortalRClient.Rmd" class="external-link"><code>vignettes/cBioPortalRClient.Rmd</code></a></small>
      <div class="hidden name"><code>cBioPortalRClient.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Please use the devel version of the <code>AnVIL</code> Bioconductor
package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">cBioPortalData</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">AnVIL</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The cBioPortal for Cancer Genomics <a href="https://cbioportal.org" class="external-link">website</a> is a great resource for
interactive exploration of study datasets. However, it does not easily
allow the analyst to obtain and further analyze the data.</p>
<p>We’ve developed the <code>cBioPortalData</code> package to fill this
need to programmatically access the data resources available on the
cBioPortal.</p>
<p>The <code>cBioPortalData</code> package provides an R interface for
accessing the cBioPortal study data within the Bioconductor
ecosystem.</p>
<p>It downloads study data from the cBioPortal API (<a href="https://cbioportal.org/api" class="external-link uri">https://cbioportal.org/api</a>) and uses Bioconductor
infrastructure to cache and represent the data.</p>
<p>We use the <a href="https://dx.doi.org/10.1158/0008-5472.CAN-17-0344" class="external-link"><code>MultiAssayExperiment</code></a>
(<span class="citation">@Ramos2017-er</span>) package to integrate,
represent, and coordinate multiple experiments for the studies availble
in the cBioPortal. This package in conjunction with
<code>curatedTCGAData</code> give access to a large trove of publicly
available bioinformatic data. Please see our publication <a href="https://dx.doi.org/10.1200/CCI.19.00119" class="external-link">here</a> (<span class="citation">@Ramos2020-ya</span>).</p>
<p>We demonstrate common use cases of <code>cBioPortalData</code> and
<code>curatedTCGAData</code> during Bioconductor conference <a href="https://waldronlab.io/MultiAssayWorkshop/" class="external-link">workshops</a>.</p>
</div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>This vignette is for users / developers who would like to learn more
about the available data in <code>cBioPortalData</code> and to learn how
to hit other endpoints in the cBioPortal API implementation. The
functionality demonstrated here is used internally by the package to
create integrative representations of study datasets.</p>
<p>Note. To avoid overloading the API service, the API was designed to
only query a part of the study data. Therefore, the user is required to
enter either a set of genes of interest or a gene panel identifier.</p>
</div>
<div class="section level2">
<h2 id="api-representation">API representation<a class="anchor" aria-label="anchor" href="#api-representation"></a>
</h2>
<p>Obtaining the cBioPortal API representation object</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">cbio</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortal.html">cBioPortal</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## service: cBioPortal</span></span>
<span><span class="co">## tags(); use cbioportal$&lt;tab completion&gt;:</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 65 × 3</span></span></span>
<span><span class="co">##    tag                 operation                                  summary       </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> Cancer Types        getAllCancerTypesUsingGET                  Get all cance…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> Cancer Types        getCancerTypeUsingGET                      Get a cancer …</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> Clinical Attributes fetchClinicalAttributesUsingPOST           Fetch clinica…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> Clinical Attributes getAllClinicalAttributesInStudyUsingGET    Get all clini…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> Clinical Attributes getAllClinicalAttributesUsingGET           Get all clini…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> Clinical Attributes getClinicalAttributeInStudyUsingGET        Get specified…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> Clinical Data       fetchAllClinicalDataInStudyUsingPOST       Fetch clinica…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> Clinical Data       fetchClinicalDataUsingPOST                 Fetch clinica…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> Clinical Data       getAllClinicalDataInStudyUsingGET          Get all clini…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> Clinical Data       getAllClinicalDataOfPatientInStudyUsingGET Get all clini…</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 55 more rows</span></span></span>
<span><span class="co">## tag values:</span></span>
<span><span class="co">##   Cancer Types, Clinical Attributes, Clinical Data, Copy Number</span></span>
<span><span class="co">##   Segments, Discrete Copy Number Alterations, Gene Panel Data, Gene</span></span>
<span><span class="co">##   Panels, Generic Assay Data, Generic Assays, Genes, Info, Molecular</span></span>
<span><span class="co">##   Data, Molecular Profiles, Mutations, Patients, Sample Lists, Samples,</span></span>
<span><span class="co">##   Server running status, Studies, Treatments</span></span>
<span><span class="co">## schemas():</span></span>
<span><span class="co">##   AlleleSpecificCopyNumber, AlterationFilter,</span></span>
<span><span class="co">##   AndedPatientTreatmentFilters, AndedSampleTreatmentFilters,</span></span>
<span><span class="co">##   CancerStudy</span></span>
<span><span class="co">##   # ... with 62 more elements</span></span></code></pre>
<div class="section level3">
<h3 id="operations">Operations<a class="anchor" aria-label="anchor" href="#operations"></a>
</h3>
<p>Check available tags, operations, and descriptions as a
<code>tibble</code>:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/AnVIL/man/Services.html" class="external-link">tags</a></span><span class="op">(</span><span class="va">cbio</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 65 × 3</span></span></span>
<span><span class="co">##    tag                 operation                                  summary       </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> Cancer Types        getAllCancerTypesUsingGET                  Get all cance…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> Cancer Types        getCancerTypeUsingGET                      Get a cancer …</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> Clinical Attributes fetchClinicalAttributesUsingPOST           Fetch clinica…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> Clinical Attributes getAllClinicalAttributesInStudyUsingGET    Get all clini…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> Clinical Attributes getAllClinicalAttributesUsingGET           Get all clini…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> Clinical Attributes getClinicalAttributeInStudyUsingGET        Get specified…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> Clinical Data       fetchAllClinicalDataInStudyUsingPOST       Fetch clinica…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> Clinical Data       fetchClinicalDataUsingPOST                 Fetch clinica…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> Clinical Data       getAllClinicalDataInStudyUsingGET          Get all clini…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> Clinical Data       getAllClinicalDataOfPatientInStudyUsingGET Get all clini…</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 55 more rows</span></span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/AnVIL/man/Services.html" class="external-link">tags</a></span><span class="op">(</span><span class="va">cbio</span><span class="op">)</span><span class="op">$</span><span class="va">operation</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "getAllCancerTypesUsingGET"              </span></span>
<span><span class="co">## [2] "getCancerTypeUsingGET"                  </span></span>
<span><span class="co">## [3] "fetchClinicalAttributesUsingPOST"       </span></span>
<span><span class="co">## [4] "getAllClinicalAttributesInStudyUsingGET"</span></span>
<span><span class="co">## [5] "getAllClinicalAttributesUsingGET"       </span></span>
<span><span class="co">## [6] "getClinicalAttributeInStudyUsingGET"</span></span></code></pre>
<div class="section level4">
<h4 id="searching-through-the-api">Searching through the API<a class="anchor" aria-label="anchor" href="#searching-through-the-api"></a>
</h4>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">searchOps</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"clinical"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "getAllClinicalAttributesUsingGET"          </span></span>
<span><span class="co">## [2] "fetchClinicalAttributesUsingPOST"          </span></span>
<span><span class="co">## [3] "fetchClinicalDataUsingPOST"                </span></span>
<span><span class="co">## [4] "getAllClinicalAttributesInStudyUsingGET"   </span></span>
<span><span class="co">## [5] "getClinicalAttributeInStudyUsingGET"       </span></span>
<span><span class="co">## [6] "getAllClinicalDataInStudyUsingGET"         </span></span>
<span><span class="co">## [7] "fetchAllClinicalDataInStudyUsingPOST"      </span></span>
<span><span class="co">## [8] "getAllClinicalDataOfPatientInStudyUsingGET"</span></span>
<span><span class="co">## [9] "getAllClinicalDataOfSampleInStudyUsingGET"</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="studies">Studies<a class="anchor" aria-label="anchor" href="#studies"></a>
</h3>
<p>Get the list of studies available:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">getStudies</a></span><span class="op">(</span><span class="va">cbio</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 399 × 13</span></span></span>
<span><span class="co">##    name          description publicStudy pmid  citation groups status importDate</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> Adenoid Cyst… Whole exom… TRUE        2609… Martelo… ACYC;…      0 2023-12-0…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> Adenoid Cyst… Whole-exom… TRUE        2368… Ho et a… ACYC;…      0 2023-12-0…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> Adenoid Cyst… Targeted S… TRUE        2441… Ross et… ACYC;…      0 2023-12-0…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> Adenoid Cyst… Whole-geno… TRUE        2686… Rettig … ACYC;…      0 2023-12-0…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> Adenoid Cyst… WGS of 21 … TRUE        2663… Mitani … ACYC;…      0 2023-12-0…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> Adenoid Cyst… Whole-geno… TRUE        2682… Drier e… ACYC        0 2023-12-0…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> Adenoid Cyst… Whole exom… TRUE        2377… Stephen… ACYC;…      0 2023-12-0…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> Adenoid Cyst… Multi-Inst… TRUE        3148… Allen e… ACYC;…      0 2023-12-0…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> Basal Cell C… Whole-exom… TRUE        2695… Bonilla… PUBLIC      0 2023-12-0…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> Acute Lympho… Comprehens… TRUE        2573… Anderss… PUBLIC      0 2023-12-0…</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 389 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 5 more variables: allSampleCount &lt;int&gt;, readPermission &lt;lgl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   studyId &lt;chr&gt;, cancerTypeId &lt;chr&gt;, referenceGenome &lt;chr&gt;</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="clinical-data">Clinical Data<a class="anchor" aria-label="anchor" href="#clinical-data"></a>
</h3>
<p>Obtain the clinical data for a particular study:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">clinicalData</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"acc_tcga"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 92 × 85</span></span></span>
<span><span class="co">##    patientId    AGE   AJCC_PATHOLOGIC_TUMOR_STAGE ATYPICAL_MITOTIC_FIGURES      </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> TCGA-OR-A5J1 58    Stage II                    Atypical Mitotic Figures Abse…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> TCGA-OR-A5J2 44    Stage IV                    Atypical Mitotic Figures Pres…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> TCGA-OR-A5J3 23    Stage III                   Atypical Mitotic Figures Abse…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> TCGA-OR-A5J4 23    Stage IV                    Atypical Mitotic Figures Abse…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> TCGA-OR-A5J5 30    Stage III                   Atypical Mitotic Figures Pres…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> TCGA-OR-A5J6 29    Stage II                    Atypical Mitotic Figures Abse…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> TCGA-OR-A5J7 30    Stage III                   Atypical Mitotic Figures Pres…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> TCGA-OR-A5J8 66    Stage III                   Atypical Mitotic Figures Pres…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> TCGA-OR-A5J9 22    Stage II                    Atypical Mitotic Figures Abse…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> TCGA-OR-A5JA 53    Stage IV                    Atypical Mitotic Figures Pres…</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 82 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 81 more variables: CAPSULAR_INVASION &lt;chr&gt;, CLIN_M_STAGE &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   CT_SCAN_PREOP_RESULTS &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   CYTOPLASM_PRESENCE_LESS_THAN_EQUAL_25_PERCENT &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   DAYS_TO_INITIAL_PATHOLOGIC_DIAGNOSIS &lt;chr&gt;, DFS_MONTHS &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   DFS_STATUS &lt;chr&gt;, DIFFUSE_ARCHITECTURE &lt;chr&gt;, ETHNICITY &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   FORM_COMPLETION_DATE &lt;chr&gt;, HISTOLOGICAL_DIAGNOSIS &lt;chr&gt;, …</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="molecular-profiles">Molecular Profiles<a class="anchor" aria-label="anchor" href="#molecular-profiles"></a>
</h3>
<p>A table of molecular profiles for a particular study can be obtained
by running the following:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mols</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortal.html">molecularProfiles</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"acc_tcga"</span><span class="op">)</span></span>
<span><span class="va">mols</span><span class="op">[[</span><span class="st">"molecularProfileId"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "acc_tcga_rppa"                                     </span></span>
<span><span class="co">## [2] "acc_tcga_rppa_Zscores"                             </span></span>
<span><span class="co">## [3] "acc_tcga_gistic"                                   </span></span>
<span><span class="co">## [4] "acc_tcga_linear_CNA"                               </span></span>
<span><span class="co">## [5] "acc_tcga_mutations"                                </span></span>
<span><span class="co">## [6] "acc_tcga_methylation_hm450"                        </span></span>
<span><span class="co">## [7] "acc_tcga_rna_seq_v2_mrna"                          </span></span>
<span><span class="co">## [8] "acc_tcga_rna_seq_v2_mrna_median_Zscores"           </span></span>
<span><span class="co">## [9] "acc_tcga_rna_seq_v2_mrna_median_all_sample_Zscores"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="molecular-profile-data">Molecular Profile Data<a class="anchor" aria-label="anchor" href="#molecular-profile-data"></a>
</h3>
<p>The data for a molecular profile can be obtained with prior knowledge
of available <code>entrezGeneIds</code>:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">molecularData</a></span><span class="op">(</span><span class="va">cbio</span>, molecularProfileIds <span class="op">=</span> <span class="st">"acc_tcga_rna_seq_v2_mrna"</span>,</span>
<span>    entrezGeneIds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>    sampleIds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TCGA-OR-A5J1-01"</span>,  <span class="st">"TCGA-OR-A5J2-01"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $acc_tcga_rna_seq_v2_mrna</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 4 × 8</span></span></span>
<span><span class="co">##   uniqueSampleKey      uniquePatientKey entrezGeneId molecularProfileId sampleId</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> VENHQS1PUi1BNUoxLTA… VENHQS1PUi1BNUo…            1 acc_tcga_rna_seq_… TCGA-OR…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> VENHQS1PUi1BNUoxLTA… VENHQS1PUi1BNUo…            2 acc_tcga_rna_seq_… TCGA-OR…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> VENHQS1PUi1BNUoyLTA… VENHQS1PUi1BNUo…            1 acc_tcga_rna_seq_… TCGA-OR…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> VENHQS1PUi1BNUoyLTA… VENHQS1PUi1BNUo…            2 acc_tcga_rna_seq_… TCGA-OR…</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 3 more variables: patientId &lt;chr&gt;, studyId &lt;chr&gt;, value &lt;dbl&gt;</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="genes">Genes<a class="anchor" aria-label="anchor" href="#genes"></a>
</h3>
<div class="section level4">
<h4 id="all-available-genes">All available genes<a class="anchor" aria-label="anchor" href="#all-available-genes"></a>
</h4>
<p>A list of all the genes provided by the API service including hugo
symbols, and entrez gene IDs can be obtained by using the
<code>geneTable</code> function:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">geneTable</a></span><span class="op">(</span><span class="va">cbio</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 1,000 × 3</span></span></span>
<span><span class="co">##    entrezGeneId hugoGeneSymbol  type </span></span>
<span><span class="co">##           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span>        -<span style="color: #BB0000; text-decoration: underline;">3</span><span style="color: #BB0000;">624</span> MIR-10A/10A     miRNA</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span>        -<span style="color: #BB0000; text-decoration: underline;">3</span><span style="color: #BB0000;">712</span> MIR-559/559     miRNA</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span>        -<span style="color: #BB0000; text-decoration: underline;">3</span><span style="color: #BB0000;">042</span> MIR-4315-2/4315 miRNA</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span>        -<span style="color: #BB0000; text-decoration: underline;">3</span><span style="color: #BB0000;">204</span> MIR-4535/4535   miRNA</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span>        -<span style="color: #BB0000; text-decoration: underline;">3</span><span style="color: #BB0000;">763</span> MIR-607/607     miRNA</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span>        -<span style="color: #BB0000; text-decoration: underline;">3</span><span style="color: #BB0000;">457</span> MIR-1269A/1269A miRNA</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span>        -<span style="color: #BB0000; text-decoration: underline;">3</span><span style="color: #BB0000;">286</span> MIR-4679-1/4679 miRNA</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span>        -<span style="color: #BB0000; text-decoration: underline;">3</span><span style="color: #BB0000;">295</span> MIR-4686/4686   miRNA</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span>        -<span style="color: #BB0000; text-decoration: underline;">3</span><span style="color: #BB0000;">054</span> MIR-4325/4325   miRNA</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span>        -<span style="color: #BB0000; text-decoration: underline;">3</span><span style="color: #BB0000;">510</span> MIR-124A-1/5P   miRNA</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 990 more rows</span></span></span></code></pre>
</div>
<div class="section level4">
<h4 id="gene-panels">Gene Panels<a class="anchor" aria-label="anchor" href="#gene-panels"></a>
</h4>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">genePanels</a></span><span class="op">(</span><span class="va">cbio</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 57 × 2</span></span></span>
<span><span class="co">##    description                                                       genePanelId</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                                             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> Targeted (27 cancer genes) sequencing of adenoid cystic carcinom… ACYC_FMI_27</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> Targeted panel of 232 genes.                                      Agilent    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> Targeted panel of 8 genes.                                        AmpliSeq   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> ARCHER-HEME gene panel (199 genes)                                ARCHER-HEM…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> ARCHER-SOLID Gene Panel (62 genes)                                ARCHER-SOL…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> Targeted sequencing of various tumor types via bait v3.           bait_v3    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> Targeted sequencing of various tumor types via bait v4.           bait_v4    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> Targeted sequencing of various tumor types via bait v5.           bait_v5    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> Targeted panel of 387 cancer-related genes.                       bcc_unige_…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> Research (CMO) IMPACT-Heme gene panel version 3.                  HemePACT_v3</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 47 more rows</span></span></span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">getGenePanel</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"IMPACT341"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 341 × 2</span></span></span>
<span><span class="co">##    entrezGeneId hugoGeneSymbol</span></span>
<span><span class="co">##           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span>           25 ABL1          </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span>        <span style="text-decoration: underline;">84</span>142 ABRAXAS1      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span>          207 AKT1          </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span>          208 AKT2          </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span>        <span style="text-decoration: underline;">10</span>000 AKT3          </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span>          238 ALK           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span>          242 ALOX12B       </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span>       <span style="text-decoration: underline;">139</span>285 AMER1         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span>          324 APC           </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span>          367 AR            </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 331 more rows</span></span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="molecular-gene-panels">Molecular Gene Panels<a class="anchor" aria-label="anchor" href="#molecular-gene-panels"></a>
</h3>
<div class="section level4">
<h4 id="genepanelmolecular">genePanelMolecular<a class="anchor" aria-label="anchor" href="#genepanelmolecular"></a>
</h4>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gprppa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortal.html">genePanelMolecular</a></span><span class="op">(</span><span class="va">cbio</span>,</span>
<span>    molecularProfileId <span class="op">=</span> <span class="st">"acc_tcga_rppa"</span>,</span>
<span>    sampleListId <span class="op">=</span> <span class="st">"acc_tcga_all"</span><span class="op">)</span></span>
<span><span class="va">gprppa</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 92 × 7</span></span></span>
<span><span class="co">##    uniqueSampleKey        uniquePatientKey molecularProfileId sampleId patientId</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> VENHQS1PUi1BNUoxLTAxO… VENHQS1PUi1BNUo… acc_tcga_rppa      TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> VENHQS1PUi1BNUoyLTAxO… VENHQS1PUi1BNUo… acc_tcga_rppa      TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> VENHQS1PUi1BNUozLTAxO… VENHQS1PUi1BNUo… acc_tcga_rppa      TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> VENHQS1PUi1BNUo0LTAxO… VENHQS1PUi1BNUo… acc_tcga_rppa      TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> VENHQS1PUi1BNUo1LTAxO… VENHQS1PUi1BNUo… acc_tcga_rppa      TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> VENHQS1PUi1BNUo2LTAxO… VENHQS1PUi1BNUo… acc_tcga_rppa      TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> VENHQS1PUi1BNUo3LTAxO… VENHQS1PUi1BNUo… acc_tcga_rppa      TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> VENHQS1PUi1BNUo4LTAxO… VENHQS1PUi1BNUo… acc_tcga_rppa      TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> VENHQS1PUi1BNUo5LTAxO… VENHQS1PUi1BNUo… acc_tcga_rppa      TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> VENHQS1PUi1BNUpBLTAxO… VENHQS1PUi1BNUp… acc_tcga_rppa      TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 82 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 2 more variables: studyId &lt;chr&gt;, profiled &lt;lgl&gt;</span></span></span></code></pre>
</div>
<div class="section level4">
<h4 id="getgenepanelmolecular">getGenePanelMolecular<a class="anchor" aria-label="anchor" href="#getgenepanelmolecular"></a>
</h4>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">getGenePanelMolecular</a></span><span class="op">(</span><span class="va">cbio</span>,</span>
<span>    molecularProfileIds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"acc_tcga_rppa"</span>, <span class="st">"acc_tcga_gistic"</span><span class="op">)</span>,</span>
<span>    sampleIds <span class="op">=</span> <span class="fu"><a href="../reference/cBioPortal.html">allSamples</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"acc_tcga"</span><span class="op">)</span><span class="op">$</span><span class="va">sampleId</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 184 × 7</span></span></span>
<span><span class="co">##    uniqueSampleKey        uniquePatientKey molecularProfileId sampleId patientId</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> VENHQS1PUi1BNUoxLTAxO… VENHQS1PUi1BNUo… acc_tcga_gistic    TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> VENHQS1PUi1BNUoyLTAxO… VENHQS1PUi1BNUo… acc_tcga_gistic    TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> VENHQS1PUi1BNUozLTAxO… VENHQS1PUi1BNUo… acc_tcga_gistic    TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> VENHQS1PUi1BNUo0LTAxO… VENHQS1PUi1BNUo… acc_tcga_gistic    TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> VENHQS1PUi1BNUo1LTAxO… VENHQS1PUi1BNUo… acc_tcga_gistic    TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> VENHQS1PUi1BNUo2LTAxO… VENHQS1PUi1BNUo… acc_tcga_gistic    TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> VENHQS1PUi1BNUo3LTAxO… VENHQS1PUi1BNUo… acc_tcga_gistic    TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> VENHQS1PUi1BNUo4LTAxO… VENHQS1PUi1BNUo… acc_tcga_gistic    TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> VENHQS1PUi1BNUo5LTAxO… VENHQS1PUi1BNUo… acc_tcga_gistic    TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> VENHQS1PUi1BNUpBLTAxO… VENHQS1PUi1BNUp… acc_tcga_gistic    TCGA-OR… TCGA-OR-…</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 174 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 2 more variables: studyId &lt;chr&gt;, profiled &lt;lgl&gt;</span></span></span></code></pre>
</div>
<div class="section level4">
<h4 id="getdatabygenes">getDataByGenes<a class="anchor" aria-label="anchor" href="#getdatabygenes"></a>
</h4>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">getDataByGenes</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"acc_tcga"</span>, genePanelId <span class="op">=</span> <span class="st">"IMPACT341"</span>,</span>
<span>    molecularProfileIds <span class="op">=</span> <span class="st">"acc_tcga_rppa"</span>, sampleListId <span class="op">=</span> <span class="st">"acc_tcga_rppa"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $acc_tcga_rppa</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 2,622 × 9</span></span></span>
<span><span class="co">##    uniqueSampleKey     uniquePatientKey entrezGeneId molecularProfileId sampleId</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> VENHQS1PUi1BNUoyLT… VENHQS1PUi1BNUo…         <span style="text-decoration: underline;">5</span>728 acc_tcga_rppa      TCGA-OR…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> VENHQS1PUi1BNUoyLT… VENHQS1PUi1BNUo…          595 acc_tcga_rppa      TCGA-OR…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> VENHQS1PUi1BNUoyLT… VENHQS1PUi1BNUo…          596 acc_tcga_rppa      TCGA-OR…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> VENHQS1PUi1BNUoyLT… VENHQS1PUi1BNUo…        <span style="text-decoration: underline;">10</span>413 acc_tcga_rppa      TCGA-OR…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> VENHQS1PUi1BNUoyLT… VENHQS1PUi1BNUo…         <span style="text-decoration: underline;">3</span>791 acc_tcga_rppa      TCGA-OR…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> VENHQS1PUi1BNUoyLT… VENHQS1PUi1BNUo…         <span style="text-decoration: underline;">7</span>157 acc_tcga_rppa      TCGA-OR…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> VENHQS1PUi1BNUoyLT… VENHQS1PUi1BNUo…          207 acc_tcga_rppa      TCGA-OR…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> VENHQS1PUi1BNUoyLT… VENHQS1PUi1BNUo…          208 acc_tcga_rppa      TCGA-OR…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> VENHQS1PUi1BNUoyLT… VENHQS1PUi1BNUo…        <span style="text-decoration: underline;">57</span>521 acc_tcga_rppa      TCGA-OR…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> VENHQS1PUi1BNUoyLT… VENHQS1PUi1BNUo…         <span style="text-decoration: underline;">2</span>064 acc_tcga_rppa      TCGA-OR…</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 2,612 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 4 more variables: patientId &lt;chr&gt;, studyId &lt;chr&gt;, value &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   hugoGeneSymbol &lt;chr&gt;</span></span></span></code></pre>
<p>It uses the <code>getAllGenesUsingGET</code> function from the
API.</p>
</div>
</div>
<div class="section level3">
<h3 id="samples">Samples<a class="anchor" aria-label="anchor" href="#samples"></a>
</h3>
<div class="section level4">
<h4 id="sample-list-identifiers">Sample List Identifiers<a class="anchor" aria-label="anchor" href="#sample-list-identifiers"></a>
</h4>
<p>To display all available sample list identifiers for a particular
study ID, one can use the <code>sampleLists</code> function:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">sampleLists</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"acc_tcga"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 9 × 5</span></span></span>
<span><span class="co">##   category                                name  description sampleListId studyId</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> all_cases_with_mrna_rnaseq_data         Samp… Samples wi… acc_tcga_rn… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> all_cases_in_study                      All … All sample… acc_tcga_all acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> all_cases_with_cna_data                 Samp… Samples wi… acc_tcga_cna acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> all_cases_with_mutation_and_cna_data    Samp… Samples wi… acc_tcga_cn… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> all_cases_with_mutation_and_cna_and_mr… Comp… Samples wi… acc_tcga_3w… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> all_cases_with_methylation_data         Samp… Samples wi… acc_tcga_me… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">7</span> all_cases_with_methylation_data         Samp… Samples wi… acc_tcga_me… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">8</span> all_cases_with_rppa_data                Samp… Samples pr… acc_tcga_rp… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">9</span> all_cases_with_mutation_data            Samp… Samples wi… acc_tcga_se… acc_tc…</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="sample-identifiers">Sample Identifiers<a class="anchor" aria-label="anchor" href="#sample-identifiers"></a>
</h4>
<p>One can obtain the barcodes / identifiers for each sample using a
specific sample list identifier, in this case we want all the copy
number alteration samples:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">samplesInSampleLists</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"acc_tcga_cna"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## CharacterList of length 1</span></span>
<span><span class="co">## [["acc_tcga_cna"]] TCGA-OR-A5J1-01 TCGA-OR-A5J2-01 ... TCGA-PK-A5HC-01</span></span></code></pre>
<p>This returns a <code>CharacterList</code> of all identifiers for each
sample list identifier input:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">samplesInSampleLists</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"acc_tcga_cna"</span>, <span class="st">"acc_tcga_cnaseq"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## CharacterList of length 2</span></span>
<span><span class="co">## [["acc_tcga_cna"]] TCGA-OR-A5J1-01 TCGA-OR-A5J2-01 ... TCGA-PK-A5HC-01</span></span>
<span><span class="co">## [["acc_tcga_cnaseq"]] TCGA-OR-A5J1-01 TCGA-OR-A5J2-01 ... TCGA-PK-A5HC-01</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="all-samples-within-a-study-id">All samples within a study ID<a class="anchor" aria-label="anchor" href="#all-samples-within-a-study-id"></a>
</h4>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">allSamples</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"acc_tcga"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 92 × 6</span></span></span>
<span><span class="co">##    uniqueSampleKey        uniquePatientKey sampleType sampleId patientId studyId</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> VENHQS1PUi1BNUoxLTAxO… VENHQS1PUi1BNUo… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> VENHQS1PUi1BNUoyLTAxO… VENHQS1PUi1BNUo… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> VENHQS1PUi1BNUozLTAxO… VENHQS1PUi1BNUo… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> VENHQS1PUi1BNUo0LTAxO… VENHQS1PUi1BNUo… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> VENHQS1PUi1BNUo1LTAxO… VENHQS1PUi1BNUo… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> VENHQS1PUi1BNUo2LTAxO… VENHQS1PUi1BNUo… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> VENHQS1PUi1BNUo3LTAxO… VENHQS1PUi1BNUo… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> VENHQS1PUi1BNUo4LTAxO… VENHQS1PUi1BNUo… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> VENHQS1PUi1BNUo5LTAxO… VENHQS1PUi1BNUo… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> VENHQS1PUi1BNUpBLTAxO… VENHQS1PUi1BNUp… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 82 more rows</span></span></span></code></pre>
</div>
<div class="section level4">
<h4 id="info-on-samples">Info on Samples<a class="anchor" aria-label="anchor" href="#info-on-samples"></a>
</h4>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">getSampleInfo</a></span><span class="op">(</span><span class="va">cbio</span>, studyId <span class="op">=</span> <span class="st">"acc_tcga"</span>,</span>
<span>    sampleListIds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"acc_tcga_rppa"</span>, <span class="st">"acc_tcga_cna"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 136 × 6</span></span></span>
<span><span class="co">##    uniqueSampleKey        uniquePatientKey sampleType sampleId patientId studyId</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> VENHQS1PUi1BNUoyLTAxO… VENHQS1PUi1BNUo… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> VENHQS1PUi1BNUozLTAxO… VENHQS1PUi1BNUo… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> VENHQS1PUi1BNUo2LTAxO… VENHQS1PUi1BNUo… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> VENHQS1PUi1BNUo3LTAxO… VENHQS1PUi1BNUo… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> VENHQS1PUi1BNUo4LTAxO… VENHQS1PUi1BNUo… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> VENHQS1PUi1BNUo5LTAxO… VENHQS1PUi1BNUo… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> VENHQS1PUi1BNUpBLTAxO… VENHQS1PUi1BNUp… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> VENHQS1PUi1BNUpQLTAxO… VENHQS1PUi1BNUp… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> VENHQS1PUi1BNUpSLTAxO… VENHQS1PUi1BNUp… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> VENHQS1PUi1BNUpTLTAxO… VENHQS1PUi1BNUp… Primary S… TCGA-OR… TCGA-OR-… acc_tc…</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 126 more rows</span></span></span></code></pre>
</div>
</div>
</div>
<div class="section level2">
<h2 id="advanced-usage">Advanced Usage<a class="anchor" aria-label="anchor" href="#advanced-usage"></a>
</h2>
<p>The <code>cBioPortal</code> API representation is not limited to the
functions provided in the package. Users who wish to make use of any of
the endpoints provided by the API specification should use the dollar
sign <code>$</code> function to access the endpoints.</p>
<p>First the user should see the input for a particular endpoint as
detailed in the API:</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cbio</span><span class="op">$</span><span class="va">getGeneUsingGET</span></span></code></pre></div>
<pre><code><span><span class="co">## getGeneUsingGET </span></span>
<span><span class="co">## Get a gene </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Parameters:</span></span>
<span><span class="co">##   geneId (string)</span></span>
<span><span class="co">##     Entrez Gene ID or Hugo Gene Symbol e.g. 1 or A1BG</span></span></code></pre>
<p>Then the user can provide such input:</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">resp</span> <span class="op">&lt;-</span> <span class="va">cbio</span><span class="op">$</span><span class="fu">getGeneUsingGET</span><span class="op">(</span><span class="st">"BRCA1"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Response [https://www.cbioportal.org/api/genes/BRCA1]</span></span>
<span><span class="co">##   Date: 2024-02-02 20:32</span></span>
<span><span class="co">##   Status: 200</span></span>
<span><span class="co">##   Content-Type: application/json</span></span>
<span><span class="co">##   Size: 69 B</span></span></code></pre>
<p>which will require the user to ‘translate’ the response using
<code><a href="https://httr.r-lib.org/reference/content.html" class="external-link">httr::content</a></code>:</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">httr</span><span class="fu">::</span><span class="fu"><a href="https://httr.r-lib.org/reference/content.html" class="external-link">content</a></span><span class="op">(</span><span class="va">resp</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $entrezGeneId</span></span>
<span><span class="co">## [1] 672</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $hugoGeneSymbol</span></span>
<span><span class="co">## [1] "BRCA1"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $type</span></span>
<span><span class="co">## [1] "protein-coding"</span></span></code></pre>
<div class="section level3">
<h3 id="clearing-the-cache">Clearing the cache<a class="anchor" aria-label="anchor" href="#clearing-the-cache"></a>
</h3>
<p>For users who wish to clear the entire <code>cBioPortalData</code>
cache, it is recommended that they use:</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="st">"~/.cache/cBioPortalData/"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="sessioninfo">sessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Etc/UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] cBioPortalData_2.14.2       MultiAssayExperiment_1.28.0</span></span>
<span><span class="co">##  [3] SummarizedExperiment_1.32.0 Biobase_2.62.0             </span></span>
<span><span class="co">##  [5] GenomicRanges_1.54.1        GenomeInfoDb_1.38.5        </span></span>
<span><span class="co">##  [7] IRanges_2.36.0              S4Vectors_0.40.2           </span></span>
<span><span class="co">##  [9] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      </span></span>
<span><span class="co">## [11] matrixStats_1.2.0           AnVIL_1.14.1               </span></span>
<span><span class="co">## [13] dplyr_1.1.4                 BiocStyle_2.30.0           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] jsonlite_1.8.8            magrittr_2.0.3           </span></span>
<span><span class="co">##   [3] GenomicFeatures_1.54.3    rmarkdown_2.25           </span></span>
<span><span class="co">##   [5] fs_1.6.3                  BiocIO_1.12.0            </span></span>
<span><span class="co">##   [7] zlibbioc_1.48.0           ragg_1.2.7               </span></span>
<span><span class="co">##   [9] vctrs_0.6.5               memoise_2.0.1            </span></span>
<span><span class="co">##  [11] Rsamtools_2.18.0          RCurl_1.98-1.14          </span></span>
<span><span class="co">##  [13] htmltools_0.5.7           S4Arrays_1.2.0           </span></span>
<span><span class="co">##  [15] progress_1.2.3            lambda.r_1.2.4           </span></span>
<span><span class="co">##  [17] curl_5.2.0                SparseArray_1.2.3        </span></span>
<span><span class="co">##  [19] sass_0.4.8                bslib_0.6.1              </span></span>
<span><span class="co">##  [21] htmlwidgets_1.6.4         desc_1.4.3               </span></span>
<span><span class="co">##  [23] futile.options_1.0.1      cachem_1.0.8             </span></span>
<span><span class="co">##  [25] GenomicAlignments_1.38.2  mime_0.12                </span></span>
<span><span class="co">##  [27] lifecycle_1.0.4           pkgconfig_2.0.3          </span></span>
<span><span class="co">##  [29] Matrix_1.6-5              R6_2.5.1                 </span></span>
<span><span class="co">##  [31] fastmap_1.1.1             GenomeInfoDbData_1.2.11  </span></span>
<span><span class="co">##  [33] shiny_1.8.0               digest_0.6.34            </span></span>
<span><span class="co">##  [35] RaggedExperiment_1.26.0   AnnotationDbi_1.64.1     </span></span>
<span><span class="co">##  [37] textshaping_0.3.7         RSQLite_2.3.5            </span></span>
<span><span class="co">##  [39] filelock_1.0.3            RTCGAToolbox_2.32.1      </span></span>
<span><span class="co">##  [41] fansi_1.0.6               RJSONIO_1.3-1.9          </span></span>
<span><span class="co">##  [43] httr_1.4.7                abind_1.4-5              </span></span>
<span><span class="co">##  [45] compiler_4.3.2            bit64_4.0.5              </span></span>
<span><span class="co">##  [47] withr_3.0.0               BiocParallel_1.36.0      </span></span>
<span><span class="co">##  [49] DBI_1.2.1                 biomaRt_2.58.2           </span></span>
<span><span class="co">##  [51] rappdirs_0.3.3            DelayedArray_0.28.0      </span></span>
<span><span class="co">##  [53] rjson_0.2.21              tools_4.3.2              </span></span>
<span><span class="co">##  [55] httpuv_1.6.14             glue_1.7.0               </span></span>
<span><span class="co">##  [57] restfulr_0.0.15           promises_1.2.1           </span></span>
<span><span class="co">##  [59] grid_4.3.2                generics_0.1.3           </span></span>
<span><span class="co">##  [61] tzdb_0.4.0                tidyr_1.3.1              </span></span>
<span><span class="co">##  [63] data.table_1.15.0         hms_1.1.3                </span></span>
<span><span class="co">##  [65] xml2_1.3.6                utf8_1.2.4               </span></span>
<span><span class="co">##  [67] XVector_0.42.0            pillar_1.9.0             </span></span>
<span><span class="co">##  [69] stringr_1.5.1             later_1.3.2              </span></span>
<span><span class="co">##  [71] BiocFileCache_2.10.1      lattice_0.22-5           </span></span>
<span><span class="co">##  [73] rtracklayer_1.62.0        bit_4.0.5                </span></span>
<span><span class="co">##  [75] tidyselect_1.2.0          Biostrings_2.70.2        </span></span>
<span><span class="co">##  [77] miniUI_0.1.1.1            knitr_1.45               </span></span>
<span><span class="co">##  [79] bookdown_0.37             futile.logger_1.4.3      </span></span>
<span><span class="co">##  [81] xfun_0.41                 DT_0.31                  </span></span>
<span><span class="co">##  [83] stringi_1.8.3             yaml_2.3.8               </span></span>
<span><span class="co">##  [85] evaluate_0.23             codetools_0.2-19         </span></span>
<span><span class="co">##  [87] tibble_3.2.1              BiocManager_1.30.22.4    </span></span>
<span><span class="co">##  [89] cli_3.6.2                 xtable_1.8-4             </span></span>
<span><span class="co">##  [91] systemfonts_1.0.5         jquerylib_0.1.4          </span></span>
<span><span class="co">##  [93] Rcpp_1.0.12               GenomicDataCommons_1.26.0</span></span>
<span><span class="co">##  [95] dbplyr_2.4.0              png_0.1-8                </span></span>
<span><span class="co">##  [97] XML_3.99-0.16.1           rapiclient_0.1.3         </span></span>
<span><span class="co">##  [99] parallel_4.3.2            TCGAutils_1.22.2         </span></span>
<span><span class="co">## [101] ellipsis_0.3.2            pkgdown_2.0.7            </span></span>
<span><span class="co">## [103] readr_2.1.5               blob_1.2.4               </span></span>
<span><span class="co">## [105] prettyunits_1.2.0         bitops_1.0-7             </span></span>
<span><span class="co">## [107] purrr_1.0.2               crayon_1.5.2             </span></span>
<span><span class="co">## [109] rlang_1.1.3               KEGGREST_1.42.0          </span></span>
<span><span class="co">## [111] rvest_1.0.3               formatR_1.14</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Levi Waldron, Marcel Ramos.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
