<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>cgdsr to cBioPortalData: Migration Tutorial • cBioPortalData</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="cgdsr to cBioPortalData: Migration Tutorial">
<meta property="og:description" content="cBioPortalData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cBioPortalData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.10.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/cBioPortalData.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cBioPortalDataErrors.html">cBioPortalData: Data Build Errors</a>
    </li>
    <li>
      <a href="../articles/cBioPortalRClient.html">cBioPortalData: API Reference Guide for Devs</a>
    </li>
    <li>
      <a href="../articles/cgdsrMigration.html">cgdsr to cBioPortalData: Migration Tutorial</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/cBioPortalData/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>cgdsr to cBioPortalData: Migration Tutorial</h1>
                        <h4 data-toc-skip class="author">Karim Mezhoud
&amp; Marcel Ramos</h4>
            
            <h4 data-toc-skip class="date">January 03, 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/cBioPortalData/blob/HEAD/vignettes/cgdsrMigration.Rmd" class="external-link"><code>vignettes/cgdsrMigration.Rmd</code></a></small>
      <div class="hidden name"><code>cgdsrMigration.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette aims to help developers migrate from the now defunct
<code>cgdsr</code> CRAN package. Note that the <code>cgdsr</code>
package code is shown for comparison but it is not guaranteed to work.
If you have questions regarding the contents, please create an issue at
the GitHub repository: <a href="https://github.com/waldronlab/cBioPortalData/issues" class="external-link uri">https://github.com/waldronlab/cBioPortalData/issues</a></p>
</div>
<div class="section level2">
<h2 id="loading-the-package">Loading the package<a class="anchor" aria-label="anchor" href="#loading-the-package"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">cBioPortalData</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="tabset tabset-fade tabset-pills" id="discovering-studies">Discovering studies<a class="anchor" aria-label="anchor" href="#discovering-studies"></a>
</h2>
<div class="section level3">
<h3 id="cbioportaldata-setup">
<code>cBioPortalData</code> setup<a class="anchor" aria-label="anchor" href="#cbioportaldata-setup"></a>
</h3>
<p>Here we show the default inputs to the cBioPortal function for
clarity.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cbio</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortal.html">cBioPortal</a></span><span class="op">(</span></span>
<span>    hostname <span class="op">=</span> <span class="st">"www.cbioportal.org"</span>,</span>
<span>    protocol <span class="op">=</span> <span class="st">"https"</span>,</span>
<span>    api. <span class="op">=</span> <span class="st">"/api/api-docs"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/cBioPortal.html">getStudies</a></span><span class="op">(</span><span class="va">cbio</span><span class="op">)</span></span></code></pre></div>
<pre><code>FALSE  [38;5;246m# A tibble: 365 × 13 [39m
FALSE    name    descr…¹ publi…² groups status impor…³ allSa…⁴ readP…⁵ studyId cance…⁶
FALSE     [3m [38;5;246m&lt;chr&gt; [39m [23m    [3m [38;5;246m&lt;chr&gt; [39m [23m    [3m [38;5;246m&lt;lgl&gt; [39m [23m    [3m [38;5;246m&lt;chr&gt; [39m [23m    [3m [38;5;246m&lt;int&gt; [39m [23m  [3m [38;5;246m&lt;chr&gt; [39m [23m      [3m [38;5;246m&lt;int&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m    [3m [38;5;246m&lt;chr&gt; [39m [23m    [3m [38;5;246m&lt;chr&gt; [39m [23m  
FALSE  [38;5;250m 1 [39m Adreno…  [38;5;246m" [39mTCGA … TRUE     [38;5;246m" [39mPUBL…      0 2022-1…      92 TRUE    acc_tc… acc    
FALSE  [38;5;250m 2 [39m Acute …  [38;5;246m" [39mCompr… TRUE     [38;5;246m" [39mPUBL…      0 2022-1…      93 TRUE    all_st… bll    
FALSE  [38;5;250m 3 [39m Hypodi…  [38;5;246m" [39mWhole… TRUE     [38;5;246m" [39m [38;5;246m" [39m          0 2022-1…      44 TRUE    all_st… myeloid
FALSE  [38;5;250m 4 [39m Adenoi…  [38;5;246m" [39mWhole… TRUE     [38;5;246m" [39mACYC…      0 2022-1…      12 TRUE    acbc_m… acbc   
FALSE  [38;5;250m 5 [39m Adenoi…  [38;5;246m" [39mTarge… TRUE     [38;5;246m" [39mACYC…      0 2022-1…      28 TRUE    acyc_f… acyc   
FALSE  [38;5;250m 6 [39m Adenoi…  [38;5;246m" [39mWhole… TRUE     [38;5;246m" [39mACYC…      0 2022-1…      25 TRUE    acyc_j… acyc   
FALSE  [38;5;250m 7 [39m Adenoi…  [38;5;246m" [39mWGS o… TRUE     [38;5;246m" [39mACYC…      0 2022-1…     102 TRUE    acyc_m… acyc   
FALSE  [38;5;250m 8 [39m Adenoi…  [38;5;246m" [39mWhole… TRUE     [38;5;246m" [39mACYC [38;5;246m" [39m      0 2022-1…      10 TRUE    acyc_m… acyc   
FALSE  [38;5;250m 9 [39m Adenoi…  [38;5;246m" [39mWhole… TRUE     [38;5;246m" [39mACYC…      0 2022-1…      24 TRUE    acyc_s… acyc   
FALSE  [38;5;250m10 [39m Acute …  [38;5;246m" [39mWhole… TRUE     [38;5;246m" [39mPUBL…      0 2022-1…      73 TRUE    all_st… bll    
FALSE  [38;5;246m# … with 355 more rows, 3 more variables: referenceGenome &lt;chr&gt;, pmid &lt;chr&gt;, [39m
FALSE  [38;5;246m#   citation &lt;chr&gt;, and abbreviated variable names ¹​description, ²​publicStudy, [39m
FALSE  [38;5;246m#   ³​importDate, ⁴​allSampleCount, ⁵​readPermission, ⁶​cancerTypeId [39m</code></pre>
<p>Note that the <code>studyId</code> column is important for further
queries.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/cBioPortal.html">getStudies</a></span><span class="op">(</span><span class="va">cbio</span><span class="op">)</span><span class="op">[[</span><span class="st">"studyId"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "acc_tcga"        "all_stjude_2015" "all_stjude_2013" "acbc_mskcc_2015"</span></span>
<span><span class="co">## [5] "acyc_fmi_2014"   "acyc_jhu_2016"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="cgdsr-setup">
<code>cgdsr</code> setup<a class="anchor" aria-label="anchor" href="#cgdsr-setup"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">cgdsr</span><span class="op">)</span></span>
<span><span class="va">cgds</span> <span class="op">&lt;-</span> <span class="fu">CGDS</span><span class="op">(</span><span class="st">"http://www.cbioportal.org/"</span><span class="op">)</span></span>
<span><span class="fu">getCancerStudies.CGDS</span><span class="op">(</span><span class="va">cgds</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 class="tabset tabset-fade tabset-pills" id="obtaining-cases">Obtaining Cases<a class="anchor" aria-label="anchor" href="#obtaining-cases"></a>
</h2>
<div class="section level3">
<h3 id="cbioportaldata-cases">
<code>cBioPortalData</code> (Cases)<a class="anchor" aria-label="anchor" href="#cbioportaldata-cases"></a>
</h3>
<div class="section level4">
<h4 id="notes">Notes<a class="anchor" aria-label="anchor" href="#notes"></a>
</h4>
<ul>
<li>Each patient is identified by a <code>patientId</code>.</li>
<li>
<code>sampleListId</code> identifies groups of
<code>patientId</code> based on profile type</li>
<li>The <code>sampleLists</code> function uses <code>studyId</code>
input to return <code>sampleListId</code>
</li>
</ul>
</div>
<div class="section level4">
<h4 id="samplelists">sampleLists<a class="anchor" aria-label="anchor" href="#samplelists"></a>
</h4>
<p>For the sample list identifiers, you can use <code>sampleLists</code>
and inspect the <code>sampleListId</code> column.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">samps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortal.html">sampleLists</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"gbm_tcga_pub"</span><span class="op">)</span></span>
<span><span class="va">samps</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"category"</span>, <span class="st">"name"</span>, <span class="st">"sampleListId"</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 15 × 3</span></span></span>
<span><span class="co">##    category                             name                             sampl…¹</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> all_cases_in_study                   All samples                      gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> other                                Expression Cluster Classical     gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> all_cases_with_cna_data              Samples with CNA data            gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> all_cases_with_mutation_and_cna_data Samples with mutation and CNA d… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> all_cases_with_mrna_array_data       Samples with mRNA data (Agilent… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> other                                Expression Cluster Mesenchymal   gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> all_cases_with_methylation_data      Samples with methylation data    gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> all_cases_with_methylation_data      Samples with methylation data (… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> all_cases_with_microrna_data         Samples with microRNA data (mic… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> other                                Expression Cluster Neural        gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">11</span> other                                Expression Cluster Proneural     gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">12</span> other                                Sequenced, No Hypermutators      gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">13</span> other                                Sequenced, Not Treated           gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">14</span> other                                Sequenced, Treated               gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">15</span> all_cases_with_mutation_data         Samples with mutation data       gbm_tc…</span></span>
<span><span class="co">## <span style="color: #949494;"># … with abbreviated variable name ¹​sampleListId</span></span></span></code></pre>
</div>
<div class="section level4">
<h4 id="samples-from-samplelists">samples from sampleLists<a class="anchor" aria-label="anchor" href="#samples-from-samplelists"></a>
</h4>
<p>It is possible to get <code>case_ids</code> directly when using the
<code>samplesInSampleLists</code> function. The function handles
multiple <code>sampleList</code> identifiers.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">samplesInSampleLists</a></span><span class="op">(</span></span>
<span>    api <span class="op">=</span> <span class="va">cbio</span>,</span>
<span>    sampleListIds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"gbm_tcga_pub_expr_classical"</span>, <span class="st">"gbm_tcga_pub_expr_mesenchymal"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## CharacterList of length 2</span></span>
<span><span class="co">## [["gbm_tcga_pub_expr_classical"]] TCGA-02-0001-01 ... TCGA-12-0615-01</span></span>
<span><span class="co">## [["gbm_tcga_pub_expr_mesenchymal"]] TCGA-02-0004-01 ... TCGA-12-0620-01</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="getsampleinfo">getSampleInfo<a class="anchor" aria-label="anchor" href="#getsampleinfo"></a>
</h4>
<p>To get more information about patients, we can query with
<code>getSampleInfo</code> function.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">getSampleInfo</a></span><span class="op">(</span>api <span class="op">=</span> <span class="va">cbio</span>,  studyId <span class="op">=</span> <span class="st">"gbm_tcga_pub"</span>, projection <span class="op">=</span> <span class="st">"SUMMARY"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 206 × 6</span></span></span>
<span><span class="co">##    uniqueSampleKey                       uniqu…¹ sampl…² sampl…³ patie…⁴ studyId</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> VENHQS0wMi0wMDAxLTAxOmdibV90Y2dhX3B1… VENHQS… Primar… TCGA-0… TCGA-0… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> VENHQS0wMi0wMDAzLTAxOmdibV90Y2dhX3B1… VENHQS… Primar… TCGA-0… TCGA-0… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> VENHQS0wMi0wMDA0LTAxOmdibV90Y2dhX3B1… VENHQS… Primar… TCGA-0… TCGA-0… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> VENHQS0wMi0wMDA2LTAxOmdibV90Y2dhX3B1… VENHQS… Primar… TCGA-0… TCGA-0… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> VENHQS0wMi0wMDA3LTAxOmdibV90Y2dhX3B1… VENHQS… Primar… TCGA-0… TCGA-0… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> VENHQS0wMi0wMDA5LTAxOmdibV90Y2dhX3B1… VENHQS… Primar… TCGA-0… TCGA-0… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> VENHQS0wMi0wMDEwLTAxOmdibV90Y2dhX3B1… VENHQS… Primar… TCGA-0… TCGA-0… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> VENHQS0wMi0wMDExLTAxOmdibV90Y2dhX3B1… VENHQS… Primar… TCGA-0… TCGA-0… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> VENHQS0wMi0wMDE0LTAxOmdibV90Y2dhX3B1… VENHQS… Primar… TCGA-0… TCGA-0… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> VENHQS0wMi0wMDE1LTAxOmdibV90Y2dhX3B1… VENHQS… Primar… TCGA-0… TCGA-0… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #949494;"># … with 196 more rows, and abbreviated variable names ¹​uniquePatientKey,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ²​sampleType, ³​sampleId, ⁴​patientId</span></span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="cgdsr-cases">
<code>cgdsr</code> (Cases)<a class="anchor" aria-label="anchor" href="#cgdsr-cases"></a>
</h3>
<div class="section level4">
<h4 id="notes-1">Notes<a class="anchor" aria-label="anchor" href="#notes-1"></a>
</h4>
<ul>
<li>‘Cases’ and ‘Patients’ are synonymous.</li>
<li>Each patient is identified by a <code>case_id</code>.</li>
<li>Queries only handle a single <code>cancerStudy</code>
identifier</li>
<li>The <code>case_list_description</code> describes the assays</li>
</ul>
</div>
<div class="section level4">
<h4 id="getcaselists-and-getclinicaldata">
<code>getCaseLists</code> and <code>getClinicalData</code><a class="anchor" aria-label="anchor" href="#getcaselists-and-getclinicaldata"></a>
</h4>
<p>We obtain the first <code>case_list_id</code> in the
<code>cgds</code> object from above and the corresponding clinical data
for that case list (<code>gbm_tcga_pub_all</code> as the case list in
this example).</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clist1</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu">getCaseLists.CGDS</span><span class="op">(</span><span class="va">cgds</span>, cancerStudy <span class="op">=</span> <span class="st">"gbm_tcga_pub"</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span>, <span class="st">"case_list_id"</span><span class="op">]</span></span>
<span></span>
<span><span class="fu">getClinicalData.CGDS</span><span class="op">(</span><span class="va">cgds</span>, <span class="va">clist1</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 class="tabset tabset-fade tabset-pills" id="obtaining-clinical-data">Obtaining Clinical Data<a class="anchor" aria-label="anchor" href="#obtaining-clinical-data"></a>
</h2>
<div class="section level3">
<h3 id="cbioportaldata-clinical">
<code>cBioPortalData</code> (Clinical)<a class="anchor" aria-label="anchor" href="#cbioportaldata-clinical"></a>
</h3>
<div class="section level4">
<h4 id="all-clinical-data">All clinical data<a class="anchor" aria-label="anchor" href="#all-clinical-data"></a>
</h4>
<p>Note that a <code>sampleListId</code> is not required when using the
<code>fetchAllClinicalDataInStudyUsingPOST</code> internal endpoint.
Data for all patients can be obtained using the
<code>clinicalData</code> function.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">clinicalData</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"gbm_tcga_pub"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 206 × 24</span></span></span>
<span><span class="co">##    patie…¹ DFS_M…² DFS_S…³ KARNO…⁴ OS_MO…⁵ OS_ST…⁶ PRETR…⁷ PRIOR…⁸ SAMPL…⁹ SEX  </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> TCGA-0… 4.5041… 1:Recu… 80.0    11.605… 1:DECE… YES     NO      1       Fema…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> TCGA-0… 1.3150… 1:Recu… 100.0   4.7342… 1:DECE… NO      NO      1       Male </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> TCGA-0… 10.323… 1:Recu… 80.0    11.342… 1:DECE… NO      NO      1       Male </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> TCGA-0… 9.9287… 1:Recu… 80.0    18.345… 1:DECE… NO      NO      1       Fema…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> TCGA-0… 17.030… 1:Recu… 80.0    23.178… 1:DECE… YES     NO      1       Fema…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> TCGA-0… 8.6794… 1:Recu… 80.0    10.586… 1:DECE… NO      NO      1       Fema…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> TCGA-0… 11.539… 1:Recu… 80.0    35.408… 1:DECE… YES     NO      1       Fema…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> TCGA-0… 4.7342… 1:Recu… 80.0    20.712… 1:DECE… NO      NO      1       Fema…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> TCGA-0… <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      100.0   82.553… 1:DECE… NO      NO      1       Male </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> TCGA-0… 14.991… 1:Recu… 80.0    20.613… 1:DECE… NO      NO      1       Male </span></span>
<span><span class="co">## <span style="color: #949494;"># … with 196 more rows, 14 more variables: sampleId &lt;chr&gt;, ACGH_DATA &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   CANCER_TYPE &lt;chr&gt;, CANCER_TYPE_DETAILED &lt;chr&gt;, COMPLETE_DATA &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   FRACTION_GENOME_ALTERED &lt;chr&gt;, MRNA_DATA &lt;chr&gt;, MUTATION_COUNT &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ONCOTREE_CODE &lt;chr&gt;, SAMPLE_TYPE &lt;chr&gt;, SEQUENCED &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   SOMATIC_STATUS &lt;chr&gt;, TMB_NONSYNONYMOUS &lt;chr&gt;, TREATMENT_STATUS &lt;chr&gt;, and</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   abbreviated variable names ¹​patientId, ²​DFS_MONTHS, ³​DFS_STATUS,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ⁴​KARNOFSKY_PERFORMANCE_SCORE, ⁵​OS_MONTHS, ⁶​OS_STATUS, …</span></span></span></code></pre>
</div>
<div class="section level4">
<h4 id="by-sample-data">By sample data<a class="anchor" aria-label="anchor" href="#by-sample-data"></a>
</h4>
<p>You can use a different endpoint to obtain data for a single sample.
First, obtain a single <code>sampleId</code> with the
<code>samplesInSampleLists</code> function.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clist1</span> <span class="op">&lt;-</span> <span class="st">"gbm_tcga_pub_all"</span></span>
<span><span class="va">samplist</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortal.html">samplesInSampleLists</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="va">clist1</span><span class="op">)</span></span>
<span><span class="va">onesample</span> <span class="op">&lt;-</span> <span class="va">samplist</span><span class="op">[[</span><span class="st">"gbm_tcga_pub_all"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">onesample</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "TCGA-02-0001-01"</span></span></code></pre>
<p>Then we use the API endpoint to retrieve the data. Note that you
would run <code><a href="https://httr.r-lib.org/reference/content.html" class="external-link">httr::content</a></code> on the output to extract the
data.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cbio</span><span class="op">$</span><span class="fu">getAllClinicalDataOfSampleInStudyUsingGET</span><span class="op">(</span></span>
<span>    sampleId <span class="op">=</span> <span class="va">onesample</span>, studyId <span class="op">=</span> <span class="st">"gbm_tcga_pub"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Response [https://www.cbioportal.org/api/studies/gbm_tcga_pub/samples/TCGA-02-0001-01/clinical-data]</span></span>
<span><span class="co">##   Date: 2023-01-03 23:48</span></span>
<span><span class="co">##   Status: 200</span></span>
<span><span class="co">##   Content-Type: application/json</span></span>
<span><span class="co">##   Size: 3.31 kB</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="cgdsr-clinical">
<code>cgdsr</code> (Clinical)<a class="anchor" aria-label="anchor" href="#cgdsr-clinical"></a>
</h3>
<div class="section level4">
<h4 id="notes-2">Notes<a class="anchor" aria-label="anchor" href="#notes-2"></a>
</h4>
<ul>
<li>
<code>getClinicalData</code> uses <code>case_list_id</code> as input
without specifying the <code>study_id</code> as case list identifiers
are unique to each study.</li>
</ul>
</div>
<div class="section level4">
<h4 id="getclinicaldata">getClinicalData<a class="anchor" aria-label="anchor" href="#getclinicaldata"></a>
</h4>
<p>We query clinical data for the
<code>gbm_tcga_pub_expr_classical</code> case list identifier which is
part of the <code>gbm_tcga_pub</code> study.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">getClinicalData.CGDS</span><span class="op">(</span>x <span class="op">=</span> <span class="va">cgds</span>,</span>
<span>    caseList <span class="op">=</span> <span class="st">"gbm_tcga_pub_expr_classical"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="clinical-data-summary">Clinical Data Summary<a class="anchor" aria-label="anchor" href="#clinical-data-summary"></a>
</h2>
<p><code>cgdsr</code> allows you to obtain clinical data for a case list
subset (54 cases with <code>gbm_tcga_pub_expr_classical</code>) and
<code>cBioPortalData</code> provides clinical data for all 206 samples
in <code>gbm_tcga_pub</code> using the <code>clinicalData</code>
function.</p>
<ul>
<li>
<code>cgdsr</code> returns a <code>data.frame</code> with
<code>sampleId</code> (TCGA.02.0009.01) but not <code>patientId</code>
(TCGA.02.0009)</li>
<li>
<code>cBioPortalData</code> returns <code>sampleId</code>
(TCGA-02-0009-01) and <code>patientId</code> (TCGA-02-0009).</li>
<li>Note the differences in identifier delimiters between the two
packages, <code>cgdsr</code> provides <code>case_id</code>s with
<code>.</code> and <code>cBioPortalData</code> returns
<code>patientId</code>s with <code>-</code>.</li>
</ul>
<p>You may be interested in other clinical data endpoints. For a list,
use the <code>searchOps</code> function.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">searchOps</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"clinical"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "getAllClinicalAttributesUsingGET"          </span></span>
<span><span class="co">## [2] "fetchClinicalAttributesUsingPOST"          </span></span>
<span><span class="co">## [3] "fetchClinicalDataUsingPOST"                </span></span>
<span><span class="co">## [4] "getAllClinicalAttributesInStudyUsingGET"   </span></span>
<span><span class="co">## [5] "getClinicalAttributeInStudyUsingGET"       </span></span>
<span><span class="co">## [6] "getAllClinicalDataInStudyUsingGET"         </span></span>
<span><span class="co">## [7] "fetchAllClinicalDataInStudyUsingPOST"      </span></span>
<span><span class="co">## [8] "getAllClinicalDataOfPatientInStudyUsingGET"</span></span>
<span><span class="co">## [9] "getAllClinicalDataOfSampleInStudyUsingGET"</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="tabset tabset-fade tabset-pills" id="molecular-or-genetic-profiles">Molecular or Genetic
Profiles<a class="anchor" aria-label="anchor" href="#molecular-or-genetic-profiles"></a>
</h2>
<div class="section level3">
<h3 id="cbioportaldata-molecularprofiles">
<code>cBioPortalData</code> (molecularProfiles)<a class="anchor" aria-label="anchor" href="#cbioportaldata-molecularprofiles"></a>
</h3>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">molecularProfiles</a></span><span class="op">(</span>api <span class="op">=</span> <span class="va">cbio</span>, studyId <span class="op">=</span> <span class="st">"gbm_tcga_pub"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 10 × 8</span></span></span>
<span><span class="co">##    molecularAlterationType datat…¹ name  descr…² showP…³ patie…⁴ molec…⁵ studyId</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> COPY_NUMBER_ALTERATION  DISCRE… Puta… Putati… TRUE    FALSE   gbm_tc… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> COPY_NUMBER_ALTERATION  DISCRE… Puta… Putati… TRUE    FALSE   gbm_tc… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> MUTATION_EXTENDED       MAF     Muta… Mutati… TRUE    FALSE   gbm_tc… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> METHYLATION             CONTIN… Meth… Methyl… FALSE   FALSE   gbm_tc… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> MRNA_EXPRESSION         CONTIN… mRNA… mRNA e… FALSE   FALSE   gbm_tc… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> MRNA_EXPRESSION         Z-SCORE mRNA… 18,698… TRUE    FALSE   gbm_tc… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> MRNA_EXPRESSION         Z-SCORE mRNA… Log-tr… TRUE    FALSE   gbm_tc… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> MRNA_EXPRESSION         CONTIN… micr… expres… FALSE   FALSE   gbm_tc… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> MRNA_EXPRESSION         Z-SCORE micr… microR… FALSE   FALSE   gbm_tc… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> MRNA_EXPRESSION         Z-SCORE mRNA… mRNA a… TRUE    FALSE   gbm_tc… gbm_tc…</span></span>
<span><span class="co">## <span style="color: #949494;"># … with abbreviated variable names ¹​datatype, ²​description,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ³​showProfileInAnalysisTab, ⁴​patientLevel, ⁵​molecularProfileId</span></span></span></code></pre>
<p>Note that we want to pull the <code>molecularProfileId</code> column
to use in other queries.</p>
</div>
<div class="section level3">
<h3 id="cgdsr-getgeneticprofiles">
<code>cgdsr</code> (getGeneticProfiles)<a class="anchor" aria-label="anchor" href="#cgdsr-getgeneticprofiles"></a>
</h3>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">getGeneticProfiles.CGDS</span><span class="op">(</span><span class="va">cgds</span>, cancerStudy <span class="op">=</span> <span class="st">"gbm_tcga_pub"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 class="tabset tabset-fade tabset-pills" id="genomic-profile-data-for-a-set-of-genes">Genomic Profile Data for a
set of genes<a class="anchor" aria-label="anchor" href="#genomic-profile-data-for-a-set-of-genes"></a>
</h2>
<div class="section level3">
<h3 id="cbioportaldata-indentify-samples-and-genes">
<code>cBioPortalData</code> (Indentify samples and genes)<a class="anchor" aria-label="anchor" href="#cbioportaldata-indentify-samples-and-genes"></a>
</h3>
<p>Currently, some conversion is needed to directly use the
<code>molecularData</code> function, if you only have Hugo symbols.
First, convert to Entrez gene IDs and then obtain all the samples in the
sample list of interest.</p>
<div class="section level4">
<h4 id="convert-hugogenesymbol-to-entrezgeneid">Convert <code>hugoGeneSymbol</code> to
<code>entrezGeneId</code><a class="anchor" aria-label="anchor" href="#convert-hugogenesymbol-to-entrezgeneid"></a>
</h4>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genetab</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortal.html">queryGeneTable</a></span><span class="op">(</span><span class="va">cbio</span>,</span>
<span>    by <span class="op">=</span> <span class="st">"hugoGeneSymbol"</span>,</span>
<span>    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NF1"</span>, <span class="st">"TP53"</span>, <span class="st">"ABL1"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">genetab</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 3 × 3</span></span></span>
<span><span class="co">##   entrezGeneId hugoGeneSymbol type          </span></span>
<span><span class="co">##          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span>         <span style="text-decoration: underline;">4</span>763 NF1            protein-coding</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span>           25 ABL1           protein-coding</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span>         <span style="text-decoration: underline;">7</span>157 TP53           protein-coding</span></span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">entrez</span> <span class="op">&lt;-</span> <span class="va">genetab</span><span class="op">[[</span><span class="st">"entrezGeneId"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="obtain-all-samples-in-study">Obtain all samples in study<a class="anchor" aria-label="anchor" href="#obtain-all-samples-in-study"></a>
</h4>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">allsamps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortal.html">samplesInSampleLists</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"gbm_tcga_pub_all"</span><span class="op">)</span></span></code></pre></div>
<p>In the next section, we will show how to use the genes and sample
identifiers to obtain the molecular profile data.</p>
</div>
</div>
<div class="section level3">
<h3 id="cgdsr-profile-data">
<code>cgdsr</code> (Profile Data)<a class="anchor" aria-label="anchor" href="#cgdsr-profile-data"></a>
</h3>
<p>The <code>getProfileData</code> function allows for straightforward
retrieval of the molecular profile data with only a case list and
genetic profile identifiers.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">getProfileData.CGDS</span><span class="op">(</span>x <span class="op">=</span> <span class="va">cgds</span>,</span>
<span>    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NF1"</span>, <span class="st">"TP53"</span>, <span class="st">"ABL1"</span><span class="op">)</span>,</span>
<span>    geneticProfiles <span class="op">=</span> <span class="st">"gbm_tcga_pub_mrna"</span>,</span>
<span>    caseList <span class="op">=</span> <span class="st">"gbm_tcga_pub_all"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="molecular-data-with-cbioportaldata">Molecular Data with <code>cBioPortalData</code><a class="anchor" aria-label="anchor" href="#molecular-data-with-cbioportaldata"></a>
</h2>
<p><code>cBioPortalData</code> provides a number of options for
retrieving molecular profile data depending on the use case. Note that
<code>molecularData</code> is mostly used internally and that the
<code>cBioPortalData</code> function is the user-friendly method for
downloading such data.</p>
<div class="section level3">
<h3 id="moleculardata">
<code>molecularData</code><a class="anchor" aria-label="anchor" href="#moleculardata"></a>
</h3>
<p>We use the translated <code>entrez</code> identifiers from above.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">molecularData</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"gbm_tcga_pub_mrna"</span>,</span>
<span>    entrezGeneIds <span class="op">=</span> <span class="va">entrez</span>, sampleIds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">allsamps</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $gbm_tcga_pub_mrna</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 618 × 8</span></span></span>
<span><span class="co">##    uniqueSampleKey      uniqu…¹ entre…² molec…³ sampl…⁴ patie…⁵ studyId    value</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> VENHQS0wMi0wMDAxLTA… VENHQS…      25 gbm_tc… TCGA-0… TCGA-0… gbm_tc… -<span style="color: #BB0000;">0.174</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> VENHQS0wMi0wMDAxLTA… VENHQS…    <span style="text-decoration: underline;">4</span>763 gbm_tc… TCGA-0… TCGA-0… gbm_tc… -<span style="color: #BB0000;">0.297</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> VENHQS0wMi0wMDAxLTA… VENHQS…    <span style="text-decoration: underline;">7</span>157 gbm_tc… TCGA-0… TCGA-0… gbm_tc…  0.621  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> VENHQS0wMi0wMDAzLTA… VENHQS…      25 gbm_tc… TCGA-0… TCGA-0… gbm_tc… -<span style="color: #BB0000;">0.177</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> VENHQS0wMi0wMDAzLTA… VENHQS…    <span style="text-decoration: underline;">4</span>763 gbm_tc… TCGA-0… TCGA-0… gbm_tc… -<span style="color: #BB0000;">0.001</span><span style="color: #BB0000; text-decoration: underline;">07</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> VENHQS0wMi0wMDAzLTA… VENHQS…    <span style="text-decoration: underline;">7</span>157 gbm_tc… TCGA-0… TCGA-0… gbm_tc…  0.006<span style="text-decoration: underline;">44</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> VENHQS0wMi0wMDA0LTA… VENHQS…      25 gbm_tc… TCGA-0… TCGA-0… gbm_tc… -<span style="color: #BB0000;">0.087</span><span style="color: #BB0000; text-decoration: underline;">8</span> </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> VENHQS0wMi0wMDA0LTA… VENHQS…    <span style="text-decoration: underline;">4</span>763 gbm_tc… TCGA-0… TCGA-0… gbm_tc… -<span style="color: #BB0000;">0.236</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> VENHQS0wMi0wMDA0LTA… VENHQS…    <span style="text-decoration: underline;">7</span>157 gbm_tc… TCGA-0… TCGA-0… gbm_tc… -<span style="color: #BB0000;">0.305</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> VENHQS0wMi0wMDA2LTA… VENHQS…      25 gbm_tc… TCGA-0… TCGA-0… gbm_tc… -<span style="color: #BB0000;">0.173</span>  </span></span>
<span><span class="co">## <span style="color: #949494;"># … with 608 more rows, and abbreviated variable names ¹​uniquePatientKey,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ²​entrezGeneId, ³​molecularProfileId, ⁴​sampleId, ⁵​patientId</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="getdatabygenes">
<code>getDataByGenes</code><a class="anchor" aria-label="anchor" href="#getdatabygenes"></a>
</h3>
<p>The <code>getDataByGenes</code> function automatically figures out
all the sample identifiers in the study and it allows Hugo and Entrez
identifiers, as well as <code>genePanelId</code> inputs.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">getDataByGenes</a></span><span class="op">(</span></span>
<span>    api <span class="op">=</span>  <span class="va">cbio</span>,</span>
<span>    studyId <span class="op">=</span> <span class="st">"gbm_tcga_pub"</span>,</span>
<span>    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NF1"</span>, <span class="st">"TP53"</span>, <span class="st">"ABL1"</span><span class="op">)</span>,</span>
<span>    by <span class="op">=</span> <span class="st">"hugoGeneSymbol"</span>,</span>
<span>    molecularProfileIds <span class="op">=</span> <span class="st">"gbm_tcga_pub_mrna"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $gbm_tcga_pub_mrna</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 618 × 10</span></span></span>
<span><span class="co">##    uniqueSamp…¹ uniqu…² entre…³ molec…⁴ sampl…⁵ patie…⁶ studyId    value hugoG…⁷</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> VENHQS0wMi0… VENHQS…      25 gbm_tc… TCGA-0… TCGA-0… gbm_tc… -<span style="color: #BB0000;">0.174</span>   ABL1   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> VENHQS0wMi0… VENHQS…    <span style="text-decoration: underline;">4</span>763 gbm_tc… TCGA-0… TCGA-0… gbm_tc… -<span style="color: #BB0000;">0.297</span>   NF1    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> VENHQS0wMi0… VENHQS…    <span style="text-decoration: underline;">7</span>157 gbm_tc… TCGA-0… TCGA-0… gbm_tc…  0.621   TP53   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> VENHQS0wMi0… VENHQS…      25 gbm_tc… TCGA-0… TCGA-0… gbm_tc… -<span style="color: #BB0000;">0.177</span>   ABL1   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> VENHQS0wMi0… VENHQS…    <span style="text-decoration: underline;">4</span>763 gbm_tc… TCGA-0… TCGA-0… gbm_tc… -<span style="color: #BB0000;">0.001</span><span style="color: #BB0000; text-decoration: underline;">07</span> NF1    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> VENHQS0wMi0… VENHQS…    <span style="text-decoration: underline;">7</span>157 gbm_tc… TCGA-0… TCGA-0… gbm_tc…  0.006<span style="text-decoration: underline;">44</span> TP53   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> VENHQS0wMi0… VENHQS…      25 gbm_tc… TCGA-0… TCGA-0… gbm_tc… -<span style="color: #BB0000;">0.087</span><span style="color: #BB0000; text-decoration: underline;">8</span>  ABL1   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> VENHQS0wMi0… VENHQS…    <span style="text-decoration: underline;">4</span>763 gbm_tc… TCGA-0… TCGA-0… gbm_tc… -<span style="color: #BB0000;">0.236</span>   NF1    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> VENHQS0wMi0… VENHQS…    <span style="text-decoration: underline;">7</span>157 gbm_tc… TCGA-0… TCGA-0… gbm_tc… -<span style="color: #BB0000;">0.305</span>   TP53   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> VENHQS0wMi0… VENHQS…      25 gbm_tc… TCGA-0… TCGA-0… gbm_tc… -<span style="color: #BB0000;">0.173</span>   ABL1   </span></span>
<span><span class="co">## <span style="color: #949494;"># … with 608 more rows, 1 more variable: type &lt;chr&gt;, and abbreviated variable</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   names ¹​uniqueSampleKey, ²​uniquePatientKey, ³​entrezGeneId,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ⁴​molecularProfileId, ⁵​sampleId, ⁶​patientId, ⁷​hugoGeneSymbol</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="cbioportaldata-the-main-end-user-function">
<code>cBioPortalData</code>: the main end-user function<a class="anchor" aria-label="anchor" href="#cbioportaldata-the-main-end-user-function"></a>
</h3>
<p>It is important to note that end users who wish to obtain the data as
easily as possible should use the main <code>cBioPortalData</code>
function:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gbm_pub</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortalData.html">cBioPortalData</a></span><span class="op">(</span></span>
<span>    api <span class="op">=</span> <span class="va">cbio</span>,</span>
<span>    studyId <span class="op">=</span> <span class="st">"gbm_tcga_pub"</span>,</span>
<span>    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NF1"</span>, <span class="st">"TP53"</span>, <span class="st">"ABL1"</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"hugoGeneSymbol"</span>,</span>
<span>    molecularProfileIds <span class="op">=</span> <span class="st">"gbm_tcga_pub_mrna"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">assay</span><span class="op">(</span><span class="va">gbm_pub</span><span class="op">[[</span><span class="st">"gbm_tcga_pub_mrna"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##      TCGA-02-0001-01 TCGA-02-0003-01 TCGA-02-0004-01 TCGA-02-0006-01</span></span>
<span><span class="co">## ABL1      -0.1744878    -0.177096729     -0.08782114      -0.1733767</span></span>
<span><span class="co">## NF1       -0.2966920    -0.001066810     -0.23626512      -0.1691507</span></span>
<span><span class="co">## TP53       0.6213171     0.006435625     -0.30507285       0.3967758</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 class="tabset tabset-fade tabset-pills" id="mutation-data">Mutation Data<a class="anchor" aria-label="anchor" href="#mutation-data"></a>
</h2>
<div class="section level3">
<h3 id="cbioportaldata-mutationdata">
<code>cBioPortalData</code> (mutationData)<a class="anchor" aria-label="anchor" href="#cbioportaldata-mutationdata"></a>
</h3>
<p>Similar to <code>molecularData</code>, mutation data can be obtained
with the <code>mutationData</code> function or the
<code>getDataByGenes</code> function.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">mutationData</a></span><span class="op">(</span></span>
<span>    api <span class="op">=</span> <span class="va">cbio</span>,</span>
<span>    molecularProfileIds <span class="op">=</span> <span class="st">"gbm_tcga_pub_mutations"</span>,</span>
<span>    entrezGeneIds <span class="op">=</span> <span class="va">entrez</span>,</span>
<span>    sampleIds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">allsamps</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $gbm_tcga_pub_mutations</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 57 × 28</span></span></span>
<span><span class="co">##    uniqueSample…¹ uniqu…² molec…³ sampl…⁴ patie…⁵ entre…⁶ studyId center mutat…⁷</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">7</span>157 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">4</span>763 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">7</span>157 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">7</span>157 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">7</span>157 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">7</span>157 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">4</span>763 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">7</span>157 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">7</span>157 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">7</span>157 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #949494;"># … with 47 more rows, 19 more variables: validationStatus &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   startPosition &lt;int&gt;, endPosition &lt;int&gt;, referenceAllele &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   proteinChange &lt;chr&gt;, mutationType &lt;chr&gt;, functionalImpactScore &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   fisValue &lt;dbl&gt;, linkXvar &lt;chr&gt;, linkPdb &lt;chr&gt;, linkMsa &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ncbiBuild &lt;chr&gt;, variantType &lt;chr&gt;, keyword &lt;chr&gt;, chr &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   variantAllele &lt;chr&gt;, refseqMrnaId &lt;chr&gt;, proteinPosStart &lt;int&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   proteinPosEnd &lt;int&gt;, and abbreviated variable names ¹​uniqueSampleKey, …</span></span></span></code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">getDataByGenes</a></span><span class="op">(</span></span>
<span>    api <span class="op">=</span> <span class="va">cbio</span>,</span>
<span>    studyId <span class="op">=</span> <span class="st">"gbm_tcga_pub"</span>,</span>
<span>    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NF1"</span>, <span class="st">"TP53"</span>, <span class="st">"ABL1"</span><span class="op">)</span>,</span>
<span>    by <span class="op">=</span> <span class="st">"hugoGeneSymbol"</span>,</span>
<span>    molecularProfileIds <span class="op">=</span> <span class="st">"gbm_tcga_pub_mutations"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $gbm_tcga_pub_mutations</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 57 × 30</span></span></span>
<span><span class="co">##    uniqueSample…¹ uniqu…² molec…³ sampl…⁴ patie…⁵ entre…⁶ studyId center mutat…⁷</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">7</span>157 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">4</span>763 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">7</span>157 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">7</span>157 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">7</span>157 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">7</span>157 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">4</span>763 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">7</span>157 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">7</span>157 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> VENHQS0wMi0wM… VENHQS… gbm_tc… TCGA-0… TCGA-0…    <span style="text-decoration: underline;">7</span>157 gbm_tc… genom… Somatic</span></span>
<span><span class="co">## <span style="color: #949494;"># … with 47 more rows, 21 more variables: validationStatus &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   startPosition &lt;int&gt;, endPosition &lt;int&gt;, referenceAllele &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   proteinChange &lt;chr&gt;, mutationType &lt;chr&gt;, functionalImpactScore &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   fisValue &lt;dbl&gt;, linkXvar &lt;chr&gt;, linkPdb &lt;chr&gt;, linkMsa &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ncbiBuild &lt;chr&gt;, variantType &lt;chr&gt;, keyword &lt;chr&gt;, chr &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   variantAllele &lt;chr&gt;, refseqMrnaId &lt;chr&gt;, proteinPosStart &lt;int&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   proteinPosEnd &lt;int&gt;, hugoGeneSymbol &lt;chr&gt;, type &lt;chr&gt;, and abbreviated …</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="cgdsr-getmutationdata">
<code>cgdsr</code> (getMutationData)<a class="anchor" aria-label="anchor" href="#cgdsr-getmutationdata"></a>
</h3>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">getMutationData.CGDS</span><span class="op">(</span></span>
<span>    x <span class="op">=</span> <span class="va">cgds</span>,</span>
<span>    caseList <span class="op">=</span> <span class="st">"getMutationData"</span>,</span>
<span>    geneticProfile <span class="op">=</span> <span class="st">"gbm_tcga_pub_mutations"</span>,</span>
<span>    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NF1"</span>, <span class="st">"TP53"</span>, <span class="st">"ABL1"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 class="tabset tabset-fade tabset-pills" id="copy-number-alteration-cna">Copy Number Alteration
(CNA)<a class="anchor" aria-label="anchor" href="#copy-number-alteration-cna"></a>
</h2>
<div class="section level3">
<h3 id="cbioportaldata-cna">
<code>cBioPortalData</code> (CNA)<a class="anchor" aria-label="anchor" href="#cbioportaldata-cna"></a>
</h3>
<p>Copy Number Alteration data can be obtained with the
<code>getDataByGenes</code> function or by the main
<code>cBioPortal</code> function.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">getDataByGenes</a></span><span class="op">(</span></span>
<span>    api <span class="op">=</span> <span class="va">cbio</span>,</span>
<span>    studyId <span class="op">=</span> <span class="st">"gbm_tcga_pub"</span>,</span>
<span>    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NF1"</span>, <span class="st">"TP53"</span>, <span class="st">"ABL1"</span><span class="op">)</span>,</span>
<span>    by <span class="op">=</span> <span class="st">"hugoGeneSymbol"</span>,</span>
<span>    molecularProfileIds <span class="op">=</span> <span class="st">"gbm_tcga_pub_cna_rae"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $gbm_tcga_pub_cna_rae</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 609 × 10</span></span></span>
<span><span class="co">##    uniqueS…¹ uniqu…² entre…³ molec…⁴ sampl…⁵ patie…⁶ studyId value hugoG…⁷ type </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> VENHQS0w… VENHQS…      25 gbm_tc… TCGA-0… TCGA-0… gbm_tc…     1 ABL1    prot…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> VENHQS0w… VENHQS…    <span style="text-decoration: underline;">4</span>763 gbm_tc… TCGA-0… TCGA-0… gbm_tc…     0 NF1     prot…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> VENHQS0w… VENHQS…    <span style="text-decoration: underline;">7</span>157 gbm_tc… TCGA-0… TCGA-0… gbm_tc…     0 TP53    prot…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> VENHQS0w… VENHQS…      25 gbm_tc… TCGA-0… TCGA-0… gbm_tc…     0 ABL1    prot…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> VENHQS0w… VENHQS…    <span style="text-decoration: underline;">4</span>763 gbm_tc… TCGA-0… TCGA-0… gbm_tc…     0 NF1     prot…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> VENHQS0w… VENHQS…    <span style="text-decoration: underline;">7</span>157 gbm_tc… TCGA-0… TCGA-0… gbm_tc…     0 TP53    prot…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> VENHQS0w… VENHQS…      25 gbm_tc… TCGA-0… TCGA-0… gbm_tc…     0 ABL1    prot…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> VENHQS0w… VENHQS…    <span style="text-decoration: underline;">4</span>763 gbm_tc… TCGA-0… TCGA-0… gbm_tc…     0 NF1     prot…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> VENHQS0w… VENHQS…    <span style="text-decoration: underline;">7</span>157 gbm_tc… TCGA-0… TCGA-0… gbm_tc…     0 TP53    prot…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> VENHQS0w… VENHQS…      25 gbm_tc… TCGA-0… TCGA-0… gbm_tc…     0 ABL1    prot…</span></span>
<span><span class="co">## <span style="color: #949494;"># … with 599 more rows, and abbreviated variable names ¹​uniqueSampleKey,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ²​uniquePatientKey, ³​entrezGeneId, ⁴​molecularProfileId, ⁵​sampleId,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ⁶​patientId, ⁷​hugoGeneSymbol</span></span></span></code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortalData.html">cBioPortalData</a></span><span class="op">(</span></span>
<span>    api <span class="op">=</span> <span class="va">cbio</span>,</span>
<span>    studyId <span class="op">=</span> <span class="st">"gbm_tcga_pub"</span>,</span>
<span>    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NF1"</span>, <span class="st">"TP53"</span>, <span class="st">"ABL1"</span><span class="op">)</span>,</span>
<span>    by <span class="op">=</span> <span class="st">"hugoGeneSymbol"</span>,</span>
<span>    molecularProfileIds <span class="op">=</span> <span class="st">"gbm_tcga_pub_cna_rae"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 3 colData rownames not in sampleMap 'primary'</span></span></code></pre>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 1 listed</span></span>
<span><span class="co">##  experiment with a user-defined name and respective class.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 1:</span></span>
<span><span class="co">##  [1] gbm_tcga_pub_cna_rae: SummarizedExperiment with 3 rows and 203 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="cgdsr-cna">
<code>cgdsr</code> (CNA)<a class="anchor" aria-label="anchor" href="#cgdsr-cna"></a>
</h3>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">getProfileData.CGDS</span><span class="op">(</span></span>
<span>    x <span class="op">=</span> <span class="va">cgds</span>,</span>
<span>    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NF1"</span>, <span class="st">"TP53"</span>, <span class="st">"ABL1"</span><span class="op">)</span>,</span>
<span>    geneticProfiles <span class="op">=</span> <span class="st">"gbm_tcga_pub_cna_rae"</span>,</span>
<span>    caseList <span class="op">=</span> <span class="st">"gbm_tcga_pub_cna"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 class="tabset tabset-fade tabset-pills" id="methylation-data">Methylation Data<a class="anchor" aria-label="anchor" href="#methylation-data"></a>
</h2>
<div class="section level3">
<h3 id="cbioportaldata-methylation">
<code>cBioPortalData</code> (Methylation)<a class="anchor" aria-label="anchor" href="#cbioportaldata-methylation"></a>
</h3>
<p>Similar to Copy Number Alteration, Methylation can be obtained by
<code>getDataByGenes</code> function or by ‘cBioPortalData’
function.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortal.html">getDataByGenes</a></span><span class="op">(</span></span>
<span>    api <span class="op">=</span> <span class="va">cbio</span>,</span>
<span>    studyId <span class="op">=</span> <span class="st">"gbm_tcga_pub"</span>,</span>
<span>    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NF1"</span>, <span class="st">"TP53"</span>, <span class="st">"ABL1"</span><span class="op">)</span>,</span>
<span>    by <span class="op">=</span> <span class="st">"hugoGeneSymbol"</span>,</span>
<span>    molecularProfileIds <span class="op">=</span> <span class="st">"gbm_tcga_pub_methylation_hm27"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $gbm_tcga_pub_methylation_hm27</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 174 × 10</span></span></span>
<span><span class="co">##    unique…¹ uniqu…² entre…³ molec…⁴ sampl…⁵ patie…⁶ studyId  value hugoG…⁷ type </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> VENHQS0… VENHQS…      25 gbm_tc… TCGA-0… TCGA-0… gbm_tc… 0.103  ABL1    prot…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> VENHQS0… VENHQS…    <span style="text-decoration: underline;">4</span>763 gbm_tc… TCGA-0… TCGA-0… gbm_tc… 0.112  NF1     prot…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> VENHQS0… VENHQS…    <span style="text-decoration: underline;">7</span>157 gbm_tc… TCGA-0… TCGA-0… gbm_tc… 0.073<span style="text-decoration: underline;">5</span> TP53    prot…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> VENHQS0… VENHQS…      25 gbm_tc… TCGA-0… TCGA-0… gbm_tc… 0.202  ABL1    prot…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> VENHQS0… VENHQS…    <span style="text-decoration: underline;">4</span>763 gbm_tc… TCGA-0… TCGA-0… gbm_tc… 0.161  NF1     prot…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> VENHQS0… VENHQS…    <span style="text-decoration: underline;">7</span>157 gbm_tc… TCGA-0… TCGA-0… gbm_tc… 0.152  TP53    prot…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> VENHQS0… VENHQS…      25 gbm_tc… TCGA-0… TCGA-0… gbm_tc… 0.179  ABL1    prot…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> VENHQS0… VENHQS…    <span style="text-decoration: underline;">4</span>763 gbm_tc… TCGA-0… TCGA-0… gbm_tc… 0.161  NF1     prot…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> VENHQS0… VENHQS…    <span style="text-decoration: underline;">7</span>157 gbm_tc… TCGA-0… TCGA-0… gbm_tc… 0.170  TP53    prot…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> VENHQS0… VENHQS…      25 gbm_tc… TCGA-0… TCGA-0… gbm_tc… 0.176  ABL1    prot…</span></span>
<span><span class="co">## <span style="color: #949494;"># … with 164 more rows, and abbreviated variable names ¹​uniqueSampleKey,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ²​uniquePatientKey, ³​entrezGeneId, ⁴​molecularProfileId, ⁵​sampleId,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ⁶​patientId, ⁷​hugoGeneSymbol</span></span></span></code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cBioPortalData.html">cBioPortalData</a></span><span class="op">(</span></span>
<span>    api <span class="op">=</span> <span class="va">cbio</span>,</span>
<span>    studyId <span class="op">=</span> <span class="st">"gbm_tcga_pub"</span>,</span>
<span>    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NF1"</span>, <span class="st">"TP53"</span>, <span class="st">"ABL1"</span><span class="op">)</span>,</span>
<span>    by <span class="op">=</span> <span class="st">"hugoGeneSymbol"</span>,</span>
<span>    molecularProfileIds <span class="op">=</span> <span class="st">"gbm_tcga_pub_methylation_hm27"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 148 colData rownames not in sampleMap 'primary'</span></span></code></pre>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 1 listed</span></span>
<span><span class="co">##  experiment with a user-defined name and respective class.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 1:</span></span>
<span><span class="co">##  [1] gbm_tcga_pub_methylation_hm27: SummarizedExperiment with 3 rows and 58 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="cgdsr-methylation">
<code>cgdsr</code> (Methylation)<a class="anchor" aria-label="anchor" href="#cgdsr-methylation"></a>
</h3>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">getProfileData.CGDS</span><span class="op">(</span></span>
<span>    x <span class="op">=</span> <span class="va">cgds</span>,</span>
<span>    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NF1"</span>, <span class="st">"TP53"</span>, <span class="st">"ABL1"</span><span class="op">)</span>,</span>
<span>    geneticProfiles <span class="op">=</span> <span class="st">"gbm_tcga_pub_methylation_hm27"</span>,</span>
<span>    caseList <span class="op">=</span> <span class="st">"gbm_tcga_pub_methylation_hm27"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="sessioninfo">sessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.2 (2022-10-31)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.1 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] cBioPortalData_2.10.3       MultiAssayExperiment_1.24.0</span></span>
<span><span class="co">##  [3] SummarizedExperiment_1.28.0 Biobase_2.58.0             </span></span>
<span><span class="co">##  [5] GenomicRanges_1.50.2        GenomeInfoDb_1.34.4        </span></span>
<span><span class="co">##  [7] IRanges_2.32.0              S4Vectors_0.36.1           </span></span>
<span><span class="co">##  [9] BiocGenerics_0.44.0         MatrixGenerics_1.10.0      </span></span>
<span><span class="co">## [11] matrixStats_0.63.0          AnVIL_1.10.1               </span></span>
<span><span class="co">## [13] dplyr_1.0.10                BiocStyle_2.26.0           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] rjson_0.2.21              ellipsis_0.3.2           </span></span>
<span><span class="co">##   [3] rprojroot_2.0.3           futile.logger_1.4.3      </span></span>
<span><span class="co">##   [5] XVector_0.38.0            fs_1.5.2                 </span></span>
<span><span class="co">##   [7] DT_0.26                   bit64_4.0.5              </span></span>
<span><span class="co">##   [9] AnnotationDbi_1.60.0      fansi_1.0.3              </span></span>
<span><span class="co">##  [11] xml2_1.3.3                codetools_0.2-18         </span></span>
<span><span class="co">##  [13] splines_4.2.2             cachem_1.0.6             </span></span>
<span><span class="co">##  [15] knitr_1.41                jsonlite_1.8.4           </span></span>
<span><span class="co">##  [17] Rsamtools_2.14.0          dbplyr_2.2.1             </span></span>
<span><span class="co">##  [19] png_0.1-8                 shiny_1.7.4              </span></span>
<span><span class="co">##  [21] BiocManager_1.30.19.5     readr_2.1.3              </span></span>
<span><span class="co">##  [23] compiler_4.2.2            httr_1.4.4               </span></span>
<span><span class="co">##  [25] assertthat_0.2.1          Matrix_1.5-3             </span></span>
<span><span class="co">##  [27] fastmap_1.1.0             limma_3.54.0             </span></span>
<span><span class="co">##  [29] cli_3.5.0                 later_1.3.0              </span></span>
<span><span class="co">##  [31] formatR_1.13              htmltools_0.5.4          </span></span>
<span><span class="co">##  [33] prettyunits_1.1.1         tools_4.2.2              </span></span>
<span><span class="co">##  [35] glue_1.6.2                GenomeInfoDbData_1.2.9   </span></span>
<span><span class="co">##  [37] rappdirs_0.3.3            Rcpp_1.0.9               </span></span>
<span><span class="co">##  [39] rapiclient_0.1.3          jquerylib_0.1.4          </span></span>
<span><span class="co">##  [41] pkgdown_2.0.7             vctrs_0.5.1              </span></span>
<span><span class="co">##  [43] Biostrings_2.66.0         RJSONIO_1.3-1.6          </span></span>
<span><span class="co">##  [45] RaggedExperiment_1.22.0   rtracklayer_1.58.0       </span></span>
<span><span class="co">##  [47] xfun_0.36                 stringr_1.5.0            </span></span>
<span><span class="co">##  [49] rvest_1.0.3               RTCGAToolbox_2.28.0      </span></span>
<span><span class="co">##  [51] mime_0.12                 miniUI_0.1.1.1           </span></span>
<span><span class="co">##  [53] lifecycle_1.0.3           restfulr_0.0.15          </span></span>
<span><span class="co">##  [55] XML_3.99-0.13             zlibbioc_1.44.0          </span></span>
<span><span class="co">##  [57] RCircos_1.2.2             ragg_1.2.4               </span></span>
<span><span class="co">##  [59] hms_1.1.2                 promises_1.2.0.1         </span></span>
<span><span class="co">##  [61] parallel_4.2.2            lambda.r_1.2.4           </span></span>
<span><span class="co">##  [63] yaml_2.3.6                curl_4.3.3               </span></span>
<span><span class="co">##  [65] memoise_2.0.1             sass_0.4.4               </span></span>
<span><span class="co">##  [67] biomaRt_2.54.0            stringi_1.7.8            </span></span>
<span><span class="co">##  [69] RSQLite_2.2.20            BiocIO_1.8.0             </span></span>
<span><span class="co">##  [71] desc_1.4.2                GenomicDataCommons_1.22.0</span></span>
<span><span class="co">##  [73] GenomicFeatures_1.50.3    filelock_1.0.2           </span></span>
<span><span class="co">##  [75] BiocParallel_1.32.5       rlang_1.0.6              </span></span>
<span><span class="co">##  [77] pkgconfig_2.0.3           systemfonts_1.0.4        </span></span>
<span><span class="co">##  [79] bitops_1.0-7              evaluate_0.19            </span></span>
<span><span class="co">##  [81] lattice_0.20-45           purrr_1.0.0              </span></span>
<span><span class="co">##  [83] GenomicAlignments_1.34.0  htmlwidgets_1.6.0        </span></span>
<span><span class="co">##  [85] bit_4.0.5                 tidyselect_1.2.0         </span></span>
<span><span class="co">##  [87] magrittr_2.0.3            bookdown_0.31            </span></span>
<span><span class="co">##  [89] R6_2.5.1                  generics_0.1.3           </span></span>
<span><span class="co">##  [91] DelayedArray_0.24.0       DBI_1.1.3                </span></span>
<span><span class="co">##  [93] withr_2.5.0               pillar_1.8.1             </span></span>
<span><span class="co">##  [95] survival_3.4-0            KEGGREST_1.38.0          </span></span>
<span><span class="co">##  [97] RCurl_1.98-1.9            tibble_3.1.8             </span></span>
<span><span class="co">##  [99] crayon_1.5.2              futile.options_1.0.1     </span></span>
<span><span class="co">## [101] utf8_1.2.2                BiocFileCache_2.6.0      </span></span>
<span><span class="co">## [103] tzdb_0.3.0                rmarkdown_2.19           </span></span>
<span><span class="co">## [105] progress_1.2.2            grid_4.2.2               </span></span>
<span><span class="co">## [107] data.table_1.14.6         blob_1.2.3               </span></span>
<span><span class="co">## [109] digest_0.6.31             xtable_1.8-4             </span></span>
<span><span class="co">## [111] tidyr_1.2.1               httpuv_1.6.7             </span></span>
<span><span class="co">## [113] textshaping_0.3.6         TCGAutils_1.18.0         </span></span>
<span><span class="co">## [115] bslib_0.4.2</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Levi Waldron, Marcel Ramos.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
